<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sudarshan A. Shetty" />


<title>Assembly diet based minimal microbiome (Db-MM10)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DbMM</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="core_microbiota_metagenomic.html">Core Microbiota</a>
</li>
<li>
  <a href="wetlab_anlaysis.html">Growth and Metabolites</a>
</li>
<li>
  <a href="dada2_processing.html">DADA2 processing</a>
</li>
<li>
  <a href="qmp_analysis.html">Microbiota profile</a>
</li>
<li>
  <a href="meta_counts.html">RNASeq processing</a>
</li>
<li>
  <a href="active_composition_metatrans.html">RNAseq Active Contribution</a>
</li>
<li>
  <a href="keeg_anlaysis_initiate.html">KEGG Analysis Setup</a>
</li>
<li>
  <a href="kegg_pathway_analysis.html">KEGG Pathway Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Assembly diet based minimal microbiome (Db-MM10)</h1>
<h3 class="subtitle">Metatranscriptomics based active contribution</h3>
<h4 class="author">Sudarshan A. Shetty</h4>
<h4 class="date">Mon Apr 05 11:24:07 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DbMM/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongNocommitsyet"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> No commits yet </a>
</p>
</div>
<div id="strongRepositoryversionstrongNocommitsyet" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .Rprofile
    Untracked:  .gitattributes
    Untracked:  .gitignore
    Untracked:  DbMM.Rproj
    Untracked:  Ocucpancy_relationship.csv
    Untracked:  README.md
    Untracked:  _workflowr.yml
    Untracked:  analysis/
    Untracked:  bin/
    Untracked:  code/
    Untracked:  data/
    Untracked:  data_raw/
    Untracked:  ko00010.Glycolysis.png
    Untracked:  ko00010.png
    Untracked:  ko00010.xml
    Untracked:  ko00640.png
    Untracked:  ko00640.propanoate_metabolism.png
    Untracked:  ko00640.xml
    Untracked:  ko00650.butanoate_metabolism.png
    Untracked:  ko00650.png
    Untracked:  ko00650.xml
    Untracked:  output/
    Untracked:  structure.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We use the output of SAMSA2 pipeline for organism hits to plot contribution of each strain to the total metatranscriptomics data.</p>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<pre class="r"><code># import other necessary packages
suppressPackageStartupMessages({
  library(DESeq2)
  library(ggplot2)
  library(ggpubr)
  library(tibble)
  library(scales)
  library(dplyr)
})

#library(optparse)</code></pre>
</div>
<div id="read-files" class="section level1">
<h1>Read files</h1>
<p>The input files are stored in <code>04_metatrans/02_species_composition</code> folder.</p>
<pre class="r"><code># GET FILE NAMES Control 24h
control_files &lt;- list.files(path = &quot;data_raw/04_metatrans/samsa_comp&quot;,
                            pattern = &quot;control_*&quot;, 
                            full.names = T, recursive = FALSE)

control_names &lt;- &quot;&quot;
for (name in control_files) {
  
  x &lt;- gsub(&quot;data_raw/04_metatrans/samsa_comp/&quot;, &quot;&quot;, name)
  x &lt;- gsub(&quot;.merged.RefSeq_annot_organism.tsv&quot;, &quot;&quot;, x)
  control_names &lt;- c(control_names, x)
}
control_names</code></pre>
<pre><code>[1] &quot;&quot;              &quot;control_B1T24&quot; &quot;control_B2T24&quot; &quot;control_B3T24&quot;</code></pre>
<pre class="r"><code>control_names_trimmed &lt;- control_names[-1]

# Experimental 48h

exp_files &lt;- list.files(path = &quot;data_raw/04_metatrans/samsa_comp&quot;,
                        pattern = &quot;experimental_*&quot;,
                        full.names = T,
                        recursive = FALSE)
exp_names = &quot;&quot;
for (name in exp_files) {
  
  x &lt;- gsub(&quot;data_raw/04_metatrans/samsa_comp/&quot;, &quot;&quot;, name)
  x &lt;- gsub(&quot;.merged.RefSeq_annot_organism.tsv&quot;, &quot;&quot;, x)
  exp_names &lt;- c(exp_names, x)
}


exp_names_trimmed &lt;- exp_names[-1]

# sanity check
if (length(exp_files) == 0 || length(control_files) == 0) {
  print (&quot;\nWARNING: No files found.  Is the directory correct?  Are the files named with &#39;control_&#39; and &#39;experimental_&#39; as prefixes?&quot;)
  stop()
}</code></pre>
<div id="read-files-1" class="section level2">
<h2>Read files</h2>
<pre class="r"><code># READ IN FILES
# loading the control table
y &lt;- 0
for (x in control_files) {
  y &lt;- y + 1
  if (y == 1) {
    control_table &lt;- read.table(file = x, header = F, quote = &quot;&quot;, sep = &quot;\t&quot;)
    colnames(control_table) = c(&quot;DELETE&quot;, x, &quot;V3&quot;)
    control_table &lt;- control_table[,c(3,2)]      }
  if (y &gt; 1) {
    temp_table &lt;- read.table(file = x, header = F, quote = &quot;&quot;, sep = &quot;\t&quot;)
    colnames(temp_table) = c(&quot;DELETE&quot;, x, &quot;V3&quot;)
    temp_table &lt;- temp_table[,c(2,3)]
    control_table &lt;- merge(control_table, temp_table, by = &quot;V3&quot;, all = T)  }
}
control_table[is.na(control_table)] &lt;- 0
rownames(control_table) = control_table$V3
control_table_trimmed &lt;- data.frame(control_table[,-1])
# this next step is for if there&#39;s only 1 control file (no replicates)
if (y == 1) {
  rownames(control_table_trimmed) = rownames(control_table)
}

# loading the experimental table
z &lt;- 0
for (x in exp_files) {
  z &lt;- z + 1
  if (z == 1) {
    exp_table &lt;- read.table(file = x, header=F, quote = &quot;&quot;, sep = &quot;\t&quot;)
    colnames(exp_table) = c(&quot;DELETE&quot;, x, &quot;V3&quot;)
    exp_table &lt;- exp_table[,c(2,3)]  }
  if (z &gt; 1) {
    temp_table &lt;- read.table(file = x, header = F, quote = &quot;&quot;, sep = &quot;\t&quot;)
    colnames(temp_table) = c(&quot;DELETE&quot;, x, &quot;V3&quot;)
    exp_table &lt;- merge(exp_table, temp_table[,c(2,3)], by = &quot;V3&quot;, all = T)  }
}
exp_table[is.na(exp_table)] &lt;- 0
rownames(exp_table) = exp_table$V3
exp_table_trimmed &lt;- exp_table[,-1]

# getting the column names simplified
colnames(control_table_trimmed) = control_names_trimmed
colnames(exp_table_trimmed) = exp_names_trimmed

# merging the two tables
complete_table &lt;- merge(control_table_trimmed, exp_table_trimmed, by=0, all = TRUE)
complete_table[is.na(complete_table)] &lt;- 0
# reducing stuff down to avoid duplicates
complete_table &lt;- aggregate(. ~  Row.names, data = complete_table, sum)
rownames(complete_table) &lt;- complete_table$Row.names
complete_table &lt;- complete_table[!(complete_table$Row.names == &quot;&quot;), ]
# removing extra Row.names column
complete_table &lt;- complete_table[,-1]
complete_table.a &lt;- complete_table

colnames(complete_table.a) </code></pre>
<pre><code>[1] &quot;control_B1T24&quot;      &quot;control_B2T24&quot;      &quot;control_B3T24&quot;     
[4] &quot;experimental_B1T48&quot; &quot;experimental_B2T48&quot; &quot;experimental_B3T48&quot;</code></pre>
<pre class="r"><code>colnames(complete_table.a) &lt;- gsub(&quot;control_&quot;, &quot;&quot;, colnames(complete_table.a))
colnames(complete_table.a) &lt;- gsub(&quot;experimental_&quot;, &quot;&quot;, colnames(complete_table.a))

#Agathobacter_rectalis 3449685
#Anaerobutyricum_soehngenii 3515670
#Bacteroides_ovatus 6549480
#Bacteroides_xylanisolvens 5912968
#Coprococcus_catus 3629509
#Eubacterium_siraeum 2664040
#Faecalibacterium_prausnitzii 3090350
#Lachnospiraceae_bacterium_7_1_58FAA 5622443
#Roseburia_intestinalis 4411370 
#Subdoligranulum_variabile 3237471</code></pre>
<pre class="r"><code>head(complete_table.a)</code></pre>
<pre><code>                             B1T24   B2T24   B3T24   B1T48   B2T48   B3T48
Agathobacter_rectalis      1189883 1533353 1994131  680440 1288745 1758023
Anaerobutyricum_soehngenii  253483  385919  100388  153621  188132  462404
Bacteroides_ovatus          396928  612919 1564701  664898  778758 1144162
Bacteroides_xylanisolvens   415784  718469 1352381  627481 1064527 1337283
Coprococcus_catus           268453  256730   97931  255466  210763  806083
Eubacterium_siraeum        1705679 2685626  562719 1248773 1228419  738649</code></pre>
</div>
<div id="correct-for-genome-size" class="section level2">
<h2>Correct for genome size</h2>
<p>Not required as these are mapped on protiens and not genomes (DNA)</p>
<pre class="r"><code>complete_table.gsize &lt;- complete_table.a

#complete_table.gsize$GenomeSize &lt;- c(3.4, 3.5, 6.5, 5.9, 3.6, 2.6,3.0, 5.6, 4.4, 3.2)
# Use genome size to normalize the reads  
complete_table.gsize &lt;- c(3.4, 3.5, 6.5, 5.9, 3.6, 2.6,3.0, 5.6, 4.4, 3.2)

#corrected_tab &lt;- ceiling((complete_table.gsize) / complete_table.gsize[, &quot;GenomeSize&quot;])
# calcualte transcripts per million  
tpm &lt;- function(counts, lengths) {
  rate &lt;- counts / lengths
  rate / sum(rate) * 1e6
}

species_abund_tpm &lt;- ceiling(tpm(complete_table.a, complete_table.gsize))
species_abund_tpm.a &lt;- as.data.frame(species_abund_tpm)</code></pre>
<pre class="r"><code>species_abund_tpm.a &lt;- as.data.frame(complete_table.a)


#DT::datatable(count.aggregate.kegg)

coldata.df &lt;- data.frame(SampleID = c(&quot;B1T24&quot;,&quot;B2T24&quot;,&quot;B3T24&quot;,&quot;B1T48&quot;,&quot;B2T48&quot;,&quot;B3T48&quot;),
                            condition = c(&quot;24h&quot;,&quot;24h&quot;,&quot;24h&quot;, &quot;48h&quot;,&quot;48h&quot;,&quot;48h&quot;))
#coldata.df
#write.table(coldata.df, &quot;tables/coldata_df.txt&quot;, sep = &quot;\t&quot;)
dds &lt;- DESeqDataSetFromMatrix(countData = species_abund_tpm.a,
                              colData = coldata.df,
                              design = ~ condition)</code></pre>
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
<pre class="r"><code>dds</code></pre>
<pre><code>class: DESeqDataSet 
dim: 10 6 
metadata(1): version
assays(1): counts
rownames(10): Agathobacter_rectalis Anaerobutyricum_soehngenii ...
  Roseburia_intestinalis Subdoligranulum_variabile
rowData names(0):
colnames(6): B1T24 B2T24 ... B2T48 B3T48
colData names(2): SampleID condition</code></pre>
<pre class="r"><code>dds &lt;- DESeq(dds,fitType= &quot;local&quot;)</code></pre>
<pre><code>estimating size factors</code></pre>
<pre><code>estimating dispersions</code></pre>
<pre><code>gene-wise dispersion estimates</code></pre>
<pre><code>mean-dispersion relationship</code></pre>
<pre><code>final dispersion estimates</code></pre>
<pre><code>fitting model and testing</code></pre>
<pre class="r"><code>res &lt;- results(dds)
spces_results &lt;- data.frame(res)
spces_results &lt;- spces_results[order(-spces_results$baseMean),]

write.table(spces_results, &quot;data/04_metatrans/tables/RNA_activity_DESeq2.txt&quot;, sep=&quot;\t&quot;)
saveRDS(spces_results, &quot;data/04_metatrans/rds/spces_results.rds&quot;)</code></pre>
<p>Plot fold change</p>
<pre class="r"><code>spces_results &lt;- spces_results %&gt;% 
  rownames_to_column(&quot;BacterialStrain&quot;) %&gt;% 
  mutate(Significant = ifelse(padj &lt;= 0.05, &quot;YES&quot;, &quot;NO&quot;))

spces_results$BacterialStrain &lt;-  gsub(&quot;Lachnospiraceae_bacterium_7_1_58FAA&quot;, &quot;Flavonifractor_plautii&quot;,
                                       spces_results$BacterialStrain )

spces_results$BacterialStrain &lt;- gsub(&quot;_&quot;, &quot; &quot;,
                                       spces_results$BacterialStrain )

p.chn &lt;- ggplot(spces_results, 
                aes(y= BacterialStrain, x= log2FoldChange)) +
  geom_col(aes(fill=Significant)) + theme_classic2(base_size = 10) + 
  scale_fill_manual(values=c(&quot;#264653&quot;, &quot;#e76f51&quot;)) +
  theme(axis.text.y = element_text(hjust=1, face = &quot;italic&quot;)) + 
  geom_text(data = data.frame(), aes(x = c(Inf, -Inf),
                                     y = c(-Inf, -Inf),
                                     hjust = c(1, 0),
                                     vjust = c(-1, -1),
                                     label = c(&quot;High 48hr&quot;,&quot;High 24hr&quot;))) +
  ylab(&quot;Bacterial Strain&quot;) 
p.chn</code></pre>
<p><img src="figure/active_composition_metatrans.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;data/04_metatrans/figs/RNA_activity_foldChange.pdf&quot;, height = 3, width = 5)</code></pre>
</div>
<div id="barplot" class="section level2">
<h2>Barplot</h2>
<pre class="r"><code>species_abund_tpm.a$BacterialStrain &lt;- rownames(species_abund_tpm.a)

species_abund_tpm_ldf &lt;- reshape::melt(species_abund_tpm.a)</code></pre>
<pre><code>Using BacterialStrain as id variables</code></pre>
<pre class="r"><code># plotting

species_abund_tpm_ldf$variable2 &lt;- species_abund_tpm_ldf$variable

species_abund_tpm_ldf$variable2 &lt;- gsub(&quot;B1T&quot;, &quot;&quot;,species_abund_tpm_ldf$variable2)
species_abund_tpm_ldf$variable2 &lt;- gsub(&quot;B2T&quot;, &quot;&quot;,species_abund_tpm_ldf$variable2)
species_abund_tpm_ldf$variable2 &lt;- gsub(&quot;B3T&quot;, &quot;&quot;,species_abund_tpm_ldf$variable2)

head(species_abund_tpm_ldf)</code></pre>
<pre><code>             BacterialStrain variable   value variable2
1      Agathobacter_rectalis    B1T24 1189883        24
2 Anaerobutyricum_soehngenii    B1T24  253483        24
3         Bacteroides_ovatus    B1T24  396928        24
4  Bacteroides_xylanisolvens    B1T24  415784        24
5          Coprococcus_catus    B1T24  268453        24
6        Eubacterium_siraeum    B1T24 1705679        24</code></pre>
<pre class="r"><code>species_abund_tpm_ldf$BacterialStrain &lt;- gsub(&quot;Lachnospiraceae_bacterium_7_1_58FAA&quot;, &quot;Flavonifractor_plautii&quot;,species_abund_tpm_ldf$BacterialStrain)

ggpubr::ggbarplot(species_abund_tpm_ldf, x =&quot;BacterialStrain&quot;, y = &quot;value&quot;, 
                  color = &quot;variable2&quot;,  
                  add = &quot;mean_se&quot;, 
                  ylab = &quot;Number of reads mapped&quot;,
                  xlab  =&quot;Bacterial strain&quot;,
                  position = position_dodge(), 
                  palette = c(&quot;#d6604d&quot;, &quot;#4d4d4d&quot;)) + 
  rotate_x_text() </code></pre>
<p><img src="figure/active_composition_metatrans.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;data/04_metatrans/figs/RNA_activity_Spps_barplot_comparison.pdf&quot;, height = 7, width = 6)</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   
[3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] dplyr_1.0.2                 scales_1.1.1               
 [3] tibble_3.0.4                ggpubr_0.4.0               
 [5] ggplot2_3.3.2               DESeq2_1.29.16             
 [7] SummarizedExperiment_1.20.0 Biobase_2.50.0             
 [9] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[11] GenomicRanges_1.42.0        GenomeInfoDb_1.26.0        
[13] IRanges_2.24.0              S4Vectors_0.28.0           
[15] BiocGenerics_0.36.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] bitops_1.0-6           fs_1.5.0               bit64_4.0.5           
 [4] RColorBrewer_1.1-2     httr_1.4.2             rprojroot_1.3-2       
 [7] tools_4.0.3            backports_1.2.0        R6_2.5.0              
[10] DBI_1.1.0              colorspace_1.4-1       withr_2.3.0           
[13] tidyselect_1.1.0       bit_4.0.4              curl_4.3              
[16] compiler_4.0.3         git2r_0.27.1           DelayedArray_0.16.0   
[19] labeling_0.4.2         genefilter_1.71.0      stringr_1.4.0         
[22] digest_0.6.27          foreign_0.8-80         rmarkdown_2.4         
[25] rio_0.5.16             XVector_0.30.0         pkgconfig_2.0.3       
[28] htmltools_0.5.1.1      readxl_1.3.1           rlang_0.4.8           
[31] rstudioapi_0.11        RSQLite_2.2.1          farver_2.0.3          
[34] generics_0.1.0         BiocParallel_1.24.0    zip_2.1.1             
[37] car_3.0-10             RCurl_1.98-1.2         magrittr_1.5          
[40] GenomeInfoDbData_1.2.4 Matrix_1.2-18          Rcpp_1.0.5            
[43] munsell_0.5.0          abind_1.4-5            lifecycle_0.2.0       
[46] stringi_1.5.3          yaml_2.2.1             carData_3.0-4         
[49] zlibbioc_1.36.0        plyr_1.8.6             grid_4.0.3            
[52] blob_1.2.1             promises_1.1.1         forcats_0.5.0         
[55] crayon_1.3.4           lattice_0.20-41        haven_2.3.1           
[58] splines_4.0.3          annotate_1.67.2        hms_0.5.3             
[61] locfit_1.5-9.4         knitr_1.30             pillar_1.4.6          
[64] ggsignif_0.6.0         geneplotter_1.67.0     XML_3.99-0.5          
[67] glue_1.4.2             evaluate_0.14          data.table_1.13.0     
[70] vctrs_0.3.4            httpuv_1.5.4           cellranger_1.1.0      
[73] gtable_0.3.0           purrr_0.3.4            tidyr_1.1.2           
[76] reshape_0.8.8          xfun_0.17              openxlsx_4.2.2        
[79] xtable_1.8-4           broom_0.7.1            rstatix_0.6.0         
[82] later_1.1.0.1          survival_3.2-7         AnnotationDbi_1.51.3  
[85] memoise_1.1.0          ellipsis_0.3.1        </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   
[3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] dplyr_1.0.2                 scales_1.1.1               
 [3] tibble_3.0.4                ggpubr_0.4.0               
 [5] ggplot2_3.3.2               DESeq2_1.29.16             
 [7] SummarizedExperiment_1.20.0 Biobase_2.50.0             
 [9] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[11] GenomicRanges_1.42.0        GenomeInfoDb_1.26.0        
[13] IRanges_2.24.0              S4Vectors_0.28.0           
[15] BiocGenerics_0.36.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] bitops_1.0-6           fs_1.5.0               bit64_4.0.5           
 [4] RColorBrewer_1.1-2     httr_1.4.2             rprojroot_1.3-2       
 [7] tools_4.0.3            backports_1.2.0        R6_2.5.0              
[10] DBI_1.1.0              colorspace_1.4-1       withr_2.3.0           
[13] tidyselect_1.1.0       bit_4.0.4              curl_4.3              
[16] compiler_4.0.3         git2r_0.27.1           DelayedArray_0.16.0   
[19] labeling_0.4.2         genefilter_1.71.0      stringr_1.4.0         
[22] digest_0.6.27          foreign_0.8-80         rmarkdown_2.4         
[25] rio_0.5.16             XVector_0.30.0         pkgconfig_2.0.3       
[28] htmltools_0.5.1.1      readxl_1.3.1           rlang_0.4.8           
[31] rstudioapi_0.11        RSQLite_2.2.1          farver_2.0.3          
[34] generics_0.1.0         BiocParallel_1.24.0    zip_2.1.1             
[37] car_3.0-10             RCurl_1.98-1.2         magrittr_1.5          
[40] GenomeInfoDbData_1.2.4 Matrix_1.2-18          Rcpp_1.0.5            
[43] munsell_0.5.0          abind_1.4-5            lifecycle_0.2.0       
[46] stringi_1.5.3          yaml_2.2.1             carData_3.0-4         
[49] zlibbioc_1.36.0        plyr_1.8.6             grid_4.0.3            
[52] blob_1.2.1             promises_1.1.1         forcats_0.5.0         
[55] crayon_1.3.4           lattice_0.20-41        haven_2.3.1           
[58] splines_4.0.3          annotate_1.67.2        hms_0.5.3             
[61] locfit_1.5-9.4         knitr_1.30             pillar_1.4.6          
[64] ggsignif_0.6.0         geneplotter_1.67.0     XML_3.99-0.5          
[67] glue_1.4.2             evaluate_0.14          data.table_1.13.0     
[70] vctrs_0.3.4            httpuv_1.5.4           cellranger_1.1.0      
[73] gtable_0.3.0           purrr_0.3.4            tidyr_1.1.2           
[76] reshape_0.8.8          xfun_0.17              openxlsx_4.2.2        
[79] xtable_1.8-4           broom_0.7.1            rstatix_0.6.0         
[82] later_1.1.0.1          survival_3.2-7         AnnotationDbi_1.51.3  
[85] memoise_1.1.0          ellipsis_0.3.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
