---
title: "Assembly diet based minimal microbiome (Db-MM10)"
subtitle: "Core microbiota anlaysis from metagenomic data"
author: "Sudarshan"
date: "`r date()`"
output:
  workflowr::wflow_html:
    toc: yes
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---
  
In this document we investigate the human gut metagenomics data to identify core species.  
Originally created on `29 July 2019`    
 
# Introduction  
We searched the `curatedMetagenomic` database for core microbiota  

## Setup
Load libraries  

```{r, warning=FALSE, message=FALSE}

#library(BiocInstaller)
#biocLite("curatedMetagenomicData")
library(RColorBrewer)
library(microbiome)
library(microbiomeutilities)
library(ggpubr)
library(dplyr)
library(glue)
library(ggtext)
library(patchwork)
```

```{r eval=FALSE}
library(curatedMetagenomicData)
```


## Download public metagenomic data  

Download the datasets for gut metagenomes from the `curatedMD` R package.  


```{r eval=FALSE}

#eh = ExperimentHub()
#myquery = query(eh, "curatedMetagenomicData")
# myquery
# View(mcols(myquery))

hmp.eset <- HMP_2012.metaphlan_bugs_list.stool()
#lecha.eset <- LeChatelierE_2013.metaphlan_bugs_list.stool()
loman.eset <- LomanNJ_2013.metaphlan_bugs_list.stool()
nielsenHB.eset <- NielsenHB_2014.metaphlan_bugs_list.stool()
obregon.eset <- `Obregon-TitoAJ_2015.metaphlan_bugs_list.stool`()
#qinN.eset <- QinN_2014.metaphlan_bugs_list.stool()
rampel.eset <- RampelliS_2015.metaphlan_bugs_list.stool()
zeller.eset <- ZellerG_2014.metaphlan_bugs_list.stool()
qinj.eset <- QinJ_2012.metaphlan_bugs_list.stool()
hann.eset <- HanniganGD_2017.metaphlan_bugs_list.stool()
beng.eset <- `Bengtsson-PalmeJ_2015.metaphlan_bugs_list.stool`()
louis.eset <- LouisS_2016.metaphlan_bugs_list.stool()
schir.eset <- SchirmerM_2016.metaphlan_bugs_list.stool()

```


## Convert to phyloseq object   

```{r eval=FALSE}

hmp.pseq1 = ExpressionSet2phyloseq( hmp.eset, relab=FALSE)
#lecha.pseq1 = ExpressionSet2phyloseq( lecha.eset, relab=FALSE)
loman.pseq1 = ExpressionSet2phyloseq( loman.eset, relab=FALSE)
neil.pseq1 = ExpressionSet2phyloseq( nielsenHB.eset, relab=FALSE)
obregon.pseq1 = ExpressionSet2phyloseq( obregon.eset, relab=FALSE)
#qinN.pseq1 = ExpressionSet2phyloseq( qinN.eset, relab=FALSE)
rampel.pseq1 = ExpressionSet2phyloseq( rampel.eset, relab=FALSE)
zeller.pseq1 = ExpressionSet2phyloseq( zeller.eset, relab=FALSE)
qinj.pseq1 = ExpressionSet2phyloseq( qinj.eset, relab=FALSE)
hann.est.pseq1 = ExpressionSet2phyloseq(hann.eset, relab=FALSE)
beng.eset.pseq1 = ExpressionSet2phyloseq(beng.eset, relab=FALSE)
louis.eset.pseq1 = ExpressionSet2phyloseq(louis.eset, relab=FALSE)
schir.eset.pseq1 = ExpressionSet2phyloseq(schir.eset, relab=FALSE)


table(meta(schir.eset.pseq1)$country)
table(meta(hann.est.pseq1)$disease)
```


## Select non-diseased samples    

```{r eval=FALSE}

unique(meta(neil.pseq1)$disease)


neil.pseq1 <- subset_samples(neil.pseq1, disease == "healthy")

unique(meta(obregon.pseq1)$disease)
obregon.pseq1 <- subset_samples(obregon.pseq1, disease == "healthy")

#unique(meta(qinN.pseq1)$disease)
#qinN.pseq1 <- subset_samples(qinN.pseq1, disease == "n")

unique(meta(rampel.pseq1)$study_condition)
rampel.pseq1 <- subset_samples(rampel.pseq1, study_condition == "control")

hann.pseq1 <- subset_samples(hann.est.pseq1, disease == "healthy")

unique(meta(zeller.pseq1)$disease)
zeller.pseq1 <- subset_samples(zeller.pseq1, disease == "healthy")


unique(meta(qinj.pseq1)$disease)
qinj.pseq1 <- subset_samples(qinj.pseq1, is.na(disease))

```


```{r, eval=FALSE}

hmp.pseq1 <- prune_taxa(taxa_sums(hmp.pseq1) > 0, hmp.pseq1)
#lecha.pseq1 <- prune_taxa(taxa_sums(lecha.pseq1) > 0, lecha.pseq1)
neil.pseq1 <- prune_taxa(taxa_sums(neil.pseq1) > 0, neil.pseq1)
obregon.pseq1 <- prune_taxa(taxa_sums(obregon.pseq1) > 0, obregon.pseq1)
#qinN.pseq1 <- prune_taxa(taxa_sums(qinN.pseq1) > 0, qinN.pseq1)
rampel.pseq1 <- prune_taxa(taxa_sums(rampel.pseq1) > 0, rampel.pseq1)
zeller.pseq1 <- prune_taxa(taxa_sums(zeller.pseq1) > 0, zeller.pseq1)
qinj.pseq1 <- prune_taxa(taxa_sums(qinj.pseq1) > 0, qinj.pseq1)
hannpseq1 = prune_taxa(taxa_sums(hann.pseq1) > 0, hann.pseq1)
beng.pseq1 = prune_taxa(taxa_sums(beng.eset.pseq1) > 0, beng.eset.pseq1)
louis.pseq1 = prune_taxa(taxa_sums(louis.eset.pseq1) > 0, louis.eset.pseq1)
schir.pseq1 = prune_taxa(taxa_sums(schir.eset.pseq1) > 0, schir.eset.pseq1)


```

## Merge all data

```{r, eval=FALSE}


ps0a <- merge_phyloseq(hmp.pseq1,neil.pseq1,obregon.pseq1, 
                       rampel.pseq1,zeller.pseq1,qinj.pseq1,
                       hannpseq1,beng.pseq1,louis.pseq1,schir.pseq1)

table(meta(ps0a)$country)
table(meta(ps0a)$disease)

saveRDS(ps0a, "data/01_exploratory/rds/ps1_raw_metagenome.rds")
```


## Clean and filter  
```{r eval=FALSE}

ps0a <- readRDS("data/01_exploratory/rds/ps1_raw_metagenome.rds")

ps0b <- subset_samples(ps0a, country != "CHN") %>% subset_samples(country != "PER") %>% subset_samples(country != "TZA")

ps0c <- subset_taxa(ps0b, Kingdom == "Bacteria")

any(taxa_sums(ps0c) < 1)

ps0d <- prune_taxa(taxa_sums(ps0c) > 1, ps0c)

any(taxa_sums(ps0d) < 1)

ps0e <- subset_taxa(ps0d, !is.na(Species))

ps1a <- tax_glom(ps0e, "Species") # time consuming step

saveRDS(ps1a, "data/01_exploratory/rds/ps1a_taxglom_species.rds")


```

The above chunks are not run in this workflowr project because the snapshot data for `curatedMD` on which the study was based is from 2018. 

```{r}

ps1a <- readRDS("data/01_exploratory/rds/ps1a_taxglom_species.rds")
any(taxa_sums(ps1a) == 1)

table(meta(ps1a)$country,meta(ps1a)$country)
table(meta(ps1a)$disease)
```

Convert to relative proportions  

```{r}

ps1a.west <- microbiome::transform(ps1a, "compositional")

#head(otu_table(ps1a.west))

min(taxa_sums(ps1a.west))

# final study detail
ps1a.west
#head(meta(ps1a.west))
table(meta(ps1a.west)$disease)
table(meta(ps1a.west)$country)

ps1a.west <- subset_samples(ps1a.west, disease == "healthy")
nsamples(ps1a.west)
```
CAN DEU DNK ESP FRA ITA NLD SWE USA 
  3  97 177  71  61  11 471  70 194 
  
healthy 
   1144  
   
## Average MGS in Western gut metagenome  
```{r fig.height=4, fig.width=8}

nspecies <- microbiome::richness(ps1a.west, "observed")
median(nspecies$observed)
summary(nspecies$observed)

hist(nspecies$observed, xlab = "Number of MGS", main="Average MGS in Western gut metagenome")
```


## Core anlaysis  

```{r}

prevalences <- seq(.05, 1, .05)
#detections <- 10^seq(log10(1e-5), log10(.2), length = 10)

detections <- c(0.00001, 0.0001,0.001, 0.01)
ps1a.core <- plot_core(ps1a.west, plot.type = "heatmap", 
                       prevalences = prevalences,
                       detections = detections,
                       colours = rev(brewer.pal(9, "RdBu")),
                       min.prevalence = .60, 
                       horizontal = F)

ps1a.core


```

Let's clean this figure.

Extract the plot data 
```{r}
ps.df <- ps1a.core$data

#unique(newdata$Taxa)

head(ps.df)
#ps.df$Prevalence
newdata <- ps.df[order(ps.df$Prevalence),] 

newdata$Taxa <- gsub("s__", "", newdata$Taxa)
head(newdata)
```


```{r}

# mark Db-MM candidates 
mmsp <- c("Coprococcus_catus","Lachnospiraceae_bacterium_7_1_58FAA","Eubacterium_hallii","Bacteroides_xylanisolvens", "Roseburia_intestinalis","Eubacterium_siraeum","Bacteroides_ovatus","Subdoligranulum_unclassified", "Faecalibacterium_prausnitzii","Eubacterium_rectale")


newdata <- newdata %>% mutate(
  color = ifelse(Taxa %in% mmsp, "#D55E00", "#797979"),
  name = glue("<i style='color:{color}'>{Taxa}</i>")
  #name = fct_reorder(name, value)
)

p <- ggplot(newdata, aes(x=DetectionThreshold, reorder(name, Prevalence), fill=Prevalence))
p <- p + geom_tile(order =TRUE) + theme(axis.text.y = element_text(face="italic"))
p <- p + xlab("Detection Threshold (Relative Abundance)") + ylab("Metagenomic species")
p.core <- p + theme_bw(base_size = 8) + #scale_x_log10() +  
  scale_fill_gradientn(colours = rev(brewer.pal(5, "RdBu"))) +
#  theme_bw(base_size=10) + 
  theme(axis.text.y = element_markdown(size = 6),
        axis.text.x = element_text(size=8, angle = 90))  + 
  ggtitle(paste0("Metagenomics Core Microbiota")) 
print(p.core)
```


Save data 
```{r}

sink(file="data/01_exploratory/tables/01_core_species.txt")
unique(p.core$data$Taxa)
sink()


ggsave("data/01_exploratory/figs/01_core_microbiota_species.tiff", height = 6, width = 6)
ggsave("data/01_exploratory/figs/01_core_microbiota_species.pdf", height = 6, width = 6)

write.csv(newdata, "data/01_exploratory/tables/01_Species_level_Core_micoribotaWest.csv")
```


## Realtive abundance-occupancy
```{r}
otu.abun = apply(otu_table(ps1a.west),1,mean)

# Calculate the frequency of each OTU across all samples
# 
otu.freq = rowSums(otu_table(ps1a.west) != 0)/nsamples(ps1a.west)

# Reassign names of phyla so we only color by the top 5 phyla and mark all others as "other"
unique(tax_table(ps1a.west)[,2])
phy <- tax_table(ps1a.west)[,2]
phyla <- phy[,"Phylum"]
#phyla = as.vector(data.frame(tax_table(ps1a.west))$Phylum)

levels(phyla) = c(levels(phyla),"other")
keephyla = c("Actinobacteria","Bacteroidetes","Firmicutes", "Proteobacteria","Verrucomicrobia")
phyla[!(phyla %in% keephyla)] = "Other"
phyla = as.vector(phyla)
phyla=as.factor(phyla)

otuabun = data.frame(abundance= otu.abun,Frequency=otu.freq,phyla)

# Use color brewer to pick a color scheme for the phyla
brew = brewer.pal(7, "Paired")

# Create a scatterplot of OTUs showing their average relative abundance and frequency 
# This plot shows how rare and abundant OTUs are distributed across the
# your study.

write.csv(otuabun, "data/01_exploratory/tables/Ocucpancy_relationship.csv")

```


```{r}
p.occ <- ggplot(otuabun, aes(x= 100*(abundance),
                             y=Frequency,color=phyla)) + 
  geom_point(aes(size = Frequency), 
             shape = 21, 
             stroke = 1, alpha = 0.6) +
    xlab("Average relative abundance (%)") + 
    ylab("OTU frequency in gut metagenomes") + 
    scale_colour_brewer("Phylum",palette="Dark2") +
    theme_bw(base_size = 8) + geom_vline(xintercept = 0, 
                            linetype="dashed", color = "grey")

# now add species
p.occ$data$species <- rownames(p.occ$data)

p.occ$data$species <- gsub("s__", "", p.occ$data$species)

write.csv(otuabun, "Ocucpancy_relationship.csv")

mmsp <- c("Coprococcus_catus","Lachnospiraceae_bacterium_7_1_58FAA","Eubacterium_hallii","Bacteroides_xylanisolvens", "Roseburia_intestinalis","Eubacterium_siraeum","Bacteroides_ovatus","Subdoligranulum_unclassified", "Faecalibacterium_prausnitzii","Eubacterium_rectale")

```

Add labels of candidate species
```{r}
p.occ2 <- p.occ + ggrepel::geom_text_repel(segment.color = "grey50", 
                                           aes(label = species, fontface=3, size=2), 
                                           data = p.occ$data[p.occ$data$species %in% mmsp,], 
                                           color = "black", size = 3,
                                           force             = 0.5,
                                           nudge_y           = -0.5,
                                           direction         = "y",
                                           hjust             = 0,
                                           segment.linetype = 1,
                                           segment.curvature = -1e-20,
                                           arrow = arrow(length = unit(0.02, "npc"))) 
                                           
p.occ2 

```


## Plot for Manuscript  

```{r fig.height=5, fig.width=10}
p.core + p.occ2 + plot_annotation(tag_levels = "a") + 
  plot_layout(widths = c(0.5,1), guides = "collect") 

ggsave("data/01_exploratory/figs/01_Figure_1_RelAb_Occupancy_Abundance_relationship.pdf", height = 5, width = 10)
ggsave("data/01_exploratory/figs/01_Figure_1_RelAb_Occupancy_Abundance_relationship.tiff", height = 5, width = 10)
ggsave("data/01_exploratory/figs/01_Figure_1_RelAb_Occupancy_Abundance_relationship.pdf", height = 5, width = 10)
```


Older messier
```{r eval=FALSE}
p.occ2 <- p.occ + ggrepel::geom_text_repel(segment.color = "grey50", aes(label = species, fontface=3), 
                                           data = p.occ$data[p.occ$data$species %in% mmsp,], 
                                           color = "black", size = 4,
                                           arrow = arrow(length = unit(0.02, "npc")), 
                                           box.padding = 1.5,
                                           nudge_x = 0.05) 

p.occ2

ggsave("output/figs/01_RelAb_Occupancy_Abundance_relationship.tiff", height = 6, width = 10)
ggsave("output/figs/01_RelAb_Occupancy_Abundance_relationship.pdf", height = 6, width = 10)


ggarrange(p.core, p.occ2, labels = c("a", "b"), ncol = 2)

ggsave("output/figs/01_Figure_1_RelAb_Occupancy_Abundance_relationship.pdf", height = 8, width = 18)
ggsave("output/figs/01_Figure_1_RelAb_Occupancy_Abundance_relationship.tiff", height = 8, width = 18)
ggsave("output/figs/01_Figure_1_RelAb_Occupancy_Abundance_relationship.pdf", height = 8, width = 18)

```

### Get an overview of abundance distribution  
```{r fig.width=8, fig.height=10}
library(patchwork)
mmsp <- c("Coprococcus_catus","Lachnospiraceae_bacterium_7_1_58FAA","Eubacterium_hallii","Bacteroides_xylanisolvens", "Roseburia_intestinalis","Eubacterium_siraeum","Bacteroides_ovatus","Subdoligranulum_unclassified", "Faecalibacterium_prausnitzii","Eubacterium_rectale")

taxa_names(ps1a.west) <- gsub("s__", "", taxa_names(ps1a.west))

ccat.plot <- plot_density(ps1a.west, variable = "Coprococcus_catus", log10 = TRUE) +
  theme_minimal()
lach.plot <- plot_density(ps1a.west, variable = "Lachnospiraceae_bacterium_7_1_58FAA", log10 = TRUE)+
  theme_minimal()
ehal.plot <- plot_density(ps1a.west, variable = "Eubacterium_hallii", log10 = TRUE)+
  theme_minimal()
bxyl.plot <- plot_density(ps1a.west, variable = "Bacteroides_xylanisolvens", log10 = TRUE)+
  theme_minimal()
rint.plot <- plot_density(ps1a.west, variable = "Roseburia_intestinalis", log10 = TRUE)+
  theme_minimal()
esir.plot <- plot_density(ps1a.west, variable = "Eubacterium_siraeum", log10 = TRUE)+
  theme_minimal()
bova.plot <- plot_density(ps1a.west, variable = "Bacteroides_ovatus", log10 = TRUE)+
  theme_minimal()
svar.plot <- plot_density(ps1a.west, variable = "Subdoligranulum_unclassified", log10 = TRUE)+
  theme_minimal()
fpra.plot <- plot_density(ps1a.west, variable = "Faecalibacterium_prausnitzii", log10 = TRUE)+
  theme_minimal()
erec.plot <- plot_density(ps1a.west, variable = "Eubacterium_rectale", log10 = TRUE)+
  theme_minimal()

(ccat.plot + lach.plot) / (ehal.plot + bxyl.plot) / (rint.plot + esir.plot) / (bova.plot +svar.plot) / (fpra.plot + erec.plot)
```


```{r eval=FALSE}
sessionInfo()

```



