---
title: "Assembly diet based minimal microbiome (Db-MM10)"
subtitle: "Species function relationships via gut metabolic modules"
author: "Sudarshan A. Shetty"
date: "`r date()`"
output:
  workflowr::wflow_html:
    toc: yes
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---


# Introduction   
For disentangling the strain-level active metabolic process, transcriptomic profiles were analysed using the gut metabolic modules (GMMs) framework. The GMMs reported previously were curated to include modules based on physiological information of the 10 Db-MM strains (see methods for curation steps)  

## Setup    
```{r, message=FALSE}
suppressPackageStartupMessages({
  library(tidyverse)
  library(data.table)
  library(circlize)
  #library(omixerRpm)
  library(ggpubr)
  library(RColorBrewer)
  library(edgeR)
  library(reshape2)
  library(patchwork)
})
```


```{r eval=FALSE}
library(omixerRpm)
```


# GMM prep  

Convert the KEGG annotation and counts file in input format required by `omixerRpm` R package from [Raes lab](https://github.com/raeslab/omixer-rpm).  

```{r}

#dir.create("gmm_input")

complete_table_brite <- readRDS("data/04_metatrans/rds/complete_table_brite_filter.rds")
#head(tpm_table_brite)
#DT::datatable(complete_table_brite)
#DT::datatable(complete_table_brite_sum)
complete_table_brite_sum <- complete_table_brite %>% group_by(KO, BacterialStrain) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(complete_table_brite)

table_brite <- complete_table_brite_sum[,c(3:8)]

#head(table_brite)

rownames(table_brite) <- table_brite$KO

cpm_table_brite_count <- as.data.frame(cpm(table_brite, log = FALSE))
#head(cpm_table_brite_count)

cpm_table_brite_count$BacterialStrain <- complete_table_brite_sum$BacterialStrain
cpm_table_brite_count$KO <- complete_table_brite_sum$KO

#head(cpm_table_brite_count)
colnames(cpm_table_brite_count) <- c(
  "B1T24",
  "B2T24",
  "B3T24",
  "B1T48",
  "B2T48",
  "B3T48",
  "BacterialStrain",
  "KO")

brite_gmm <-
  as.data.frame(cpm_table_brite_count[, c("KO",
                                          "BacterialStrain",
                                          "B1T24",
                                          "B2T24",
                                          "B3T24",
                                          "B1T48",
                                          "B2T48",
                                          "B3T48")])

#head(brite_gmm)
colnames(brite_gmm) <-
  c("KO",
    "species",
    "B1T24",
    "B2T24",
    "B3T24",
    "B1T48",
    "B2T48",
    "B3T48")

head(brite_gmm)


```


```{r}

brite_gmm <-separate(brite_gmm, species, c("Bacteria", "Locus"), 
                     extra = "merge", fill = "right")

#head(brite_gmm)
brite_gmm <-separate(brite_gmm, Locus, c("Species", "Locus"), 
                     extra = "merge", fill = "right")

brite_gmm <- brite_gmm %>% unite(Species, Bacteria,Species) 

brite_gmm <- brite_gmm[,-3]
head(brite_gmm)
write.table(brite_gmm, file='data_raw/05_gmm/brite_gmm_server_filtered.tsv', 
            quote=FALSE, 
            sep='\t', col.names = NA)

```

Upload this file to `micro3` server.

## Server GMM calculations  

```{r, eval=FALSE}

############
#dat <- read.table("03_gmm_anlaysis/data/brite_gmm_server.tsv", header=T, sep="\t")
dat <- read.table("data_raw/05_gmm/brite_gmm_server_filtered.tsv", header=T, sep="\t")
#meta <- read.table("gmm/metadata.tsv", header=T, sep="\t")

db <- loadDB("GMMs.v1.07")
dat.sub <- dat[,-1]
dat.sub <- dat.sub[,c("Species","KO","B1T24", "B2T24","B3T24", "B1T48", "B2T48","B3T48")]

mods <- rpm(dat.sub, minimum.coverage=0.5, annotation = 2, module.db = db)
saveRDS(mods, "data_raw/05_gmm/mods_gmm_aa.rds")
###########

```

## Read GMMs  

```{r}
# the GMM cal was done on server due to issues with java windows pc. the mods_gmm_aa.rds file was copied to gmm 
# read file from server
#mods <- readRDS("03_gmm_anlaysis/data/mods_gmm.rds")
mods <- readRDS("data_raw/05_gmm/mods_gmm_aa.rds")

modsDF <- cbind(mods@annotation, slot(mods, "abundance"))
write.table(modsDF, file='data/05_gmm/tables/modsDF_gmm_linux.tsv', 
            quote=FALSE, sep='\t', col.names = NA)

#head(modsDF)
```


## Links Species-Functions     
```{r}

#colnames(modsDF)
modsDF2 <- modsDF
modsDF2 <- unite(modsDF, TAXMOD, "Taxon","Module")
#head(modsDF2)
rownames(modsDF2) <- modsDF2$TAXMOD
modsDF2 <- modsDF2[,-1]

#colnames(modsDF2)
#DT::datatable(modsDF2)

```


Check fold change  
```{r}

source("code/check_wilcox.r")

pval <- check_wilcoxon(modsDF2, c("48h","48h","24h","48h", "24h","24h"))

#head(pval)
summary(pval$p.value)

pval$Taxon <- modsDF$Taxon
pval$Module <- modsDF$Module
pval <- pval[,c("Module", "Taxon", "fold.change.log10", "p.value")]
mat <- pval
#DT::datatable(mat)
colnames(mat) <- c("from", "to", "Difference", "P-value")

```

```{r}
# Read names of modules
curatedGMM_names <- read.table("data_raw/05_gmm/ModulesCustom/CuratedGMM_names.txt", header=F, sep="\t")
#head(curatedGMM_names)

fil_list <- pval$Module
curatedGMM_names_filt <- filter(curatedGMM_names, curatedGMM_names$V1 %in% fil_list)

#dim(curatedGMM_names_filt)
#curatedGMM_names_filt$V1
colnames(curatedGMM_names_filt) <- c("Module", "Names", "Category")
#unique(mat$from)
#dim(pval)
mat_df <- merge(pval, curatedGMM_names_filt, by.x="Module")
#dim(mat_df)
#head(mat_df)

mat_df <- mat_df[,c("Taxon", "Names", "fold.change.log10", "p.value", "Module", "Category")]
#mat_df <- mat_df[,c("Taxon", "Module", "fold.change.log10", "p.value", "Names")]
colnames(mat_df) <- c("from", "to", "Difference", "P-value", "ModuleName", "Category")

#mat <- subset(mat_df, abs(Difference) > 0.5)
```

Specify colors  
```{r}
#head(mat_df)

# Change name 
mat_df$from <- gsub("Lachnospiraceae_bacterium", "Flavonifractor_plautii",mat_df$from)

head(mat_df)
mat_df$from <- gsub("_"," ",mat_df$from)
# set colors  
strain.colors <- c('Bacteroides ovatus' = "#1F78B4", 
                   'Bacteroides xylanisolvens' = "#B2DF8A", 
                   'Anaerobutyricum soehngenii' = "#FDBF6F", 
                   'Agathobacter rectalis' = "#33A02C", 
                   'Eubacterium siraeum' ="#FB9A99", 
                   'Faecalibacterium prausnitzii' = "#A6CEE3", 
                   'Flavonifractor plautii' = "#E31A1C", 
                   'Roseburia intestinalis' = "#CAB2D6", 
                   'Subdoligranulum variabile' = "#FF7F00", 
                   'Coprococcus catus' = "#6A3D9A")

```



## Plot Chords   
```{r warning=FALSE, message=FALSE,fig.width=16, fig.height=16}

# Split the data to positive and negative change. Anything below 0 (negative values is higher in 24h while positive are higher in 48h samples)
mat.pos <- subset(mat_df, Difference > 0)
mat.neg <- subset(mat_df, Difference < 0)

mat.pos_other <- subset(mat.pos, Category== "Other")
mat.neg_other <- subset(mat.neg, Category== "Other")
#subset(mat_df, Category== "Other" & abs(Difference) >= 0.2) %>% 
#  ggplot(aes(Difference, to)) +
#  geom_col() + facet_grid(~ from, scales = "free_y")

mat.pos_other <- mat.pos_other %>% select(-"P-value")

circos.par(start.degree = 180, gap.degree =3)
chordDiagram(mat.pos_other, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.pos))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.8, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)

############################ SAVE PLOT ####################################################
pdf("data/05_gmm/figs/GMM_positive_chord_other.pdf", height = 6, width = 10)
#par(mar=c(4.5, 0.5, 0.5, 0.5))
circos.par(start.degree = 180, gap.degree =3)
chordDiagram(mat.pos_other, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.pos))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.4, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)
dev.off()
circos.clear()

#########################################################################################

#########################################################################################
circos.par(start.degree = 180, gap.degree =3)
chordDiagram(mat.neg_other, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.neg))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.4, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)

############################ SAVE PLOT ####################################################
pdf("data/05_gmm/figs/GMM_negative_chord_other.pdf", height = 6, width = 10)
#par(mar=c(4.5, 0.5, 0.5, 0.5))
circos.par(start.degree = 180, gap.degree =3)
chordDiagram(mat.neg_other, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.neg))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.4, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)
dev.off()
circos.clear()

```

```{r warning=FALSE, message=FALSE,fig.width=16, fig.height=16}
mat.pos_aa <- subset(mat.pos, Category != "Other")
mat.neg_aa <- subset(mat.neg, Category != "Other")

mat.pos_aa <- mat.pos_aa %>% select(-"P-value")
mat.neg_aa <- mat.neg_aa %>% select(-"P-value")

circos.par(start.degree = 180, gap.degree =2)
chordDiagram(mat.pos_aa, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.pos))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.3, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)

############################ SAVE PLOT ####################################################

pdf("data/05_gmm/figs/GMM_positive_chord_AA.pdf", height = 6, width = 10)
#par(mar=c(4.5, 0.5, 0.5, 0.5))
circos.par(start.degree = 180, gap.degree =2)
chordDiagram(mat.pos_aa, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.pos))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.3, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)
dev.off()
circos.clear()

###############################################################################################

###############################################################################################
circos.par(start.degree = 180, gap.degree =2)
chordDiagram(mat.neg_aa, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.neg))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.3, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)

############################ SAVE PLOT ####################################################

pdf("data/05_gmm/figs/GMM_negative_chord_AA.pdf", height = 6, width = 10)
#par(mar=c(4.5, 0.5, 0.5, 0.5))
circos.par(start.degree = 180, gap.degree =3)
chordDiagram(mat.neg_aa, annotationTrack = "grid", big.gap = 10,
             transparency = 0.1, grid.col = strain.colors,
             preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(mat.neg))))))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 1)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 1)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 1, 
              facing = "bending.inside", niceFacing = TRUE, col = "white")
}
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, cex = 0.4, 
              facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA)
dev.off()
circos.clear()
```

# Species Function Contributions   

```{r}

#modsDF
curatedGMM_names <- read.table("data_raw/05_gmm/ModulesCustom/CuratedGMM_names.txt", header=F, sep="\t")
head(curatedGMM_names)

fil_list <- modsDF$Module
curatedGMM_names_filt <- filter(curatedGMM_names, curatedGMM_names$V1 %in% fil_list)

dim(curatedGMM_names_filt)
#curatedGMM_names_filt$V1
colnames(curatedGMM_names_filt) <- c("Module", "Names")
#unique(mat$from)
#dim(pval)
module_df <- merge(modsDF, curatedGMM_names_filt, by.x="Module")
#dim(module_df)
#head(module_df)

#count.module <- module_df %>% group_by(Names, Taxon) %>% 
#  summarise(mean(B1T24), mean(B2T24), mean(B3T24),mean(B1T48), mean(B2T48), mean(B3T48))

module_df_lng <- melt(module_df)
head(module_df_lng)
count.module <- dcast(module_df_lng, Taxon + Names + Module ~ variable, value.var="value")

#head(count.module)

colnames(count.module) <- c("Taxon", "Module", "ModuleID", "B1T24", "B2T24","B3T24","B1T48","B2T48","B3T48")
```

## All modules heatmap   

```{r fig.width=16, fig.height=20}

count.module.df1 <- reshape2::melt(count.module)

#write.csv(count.module, "03_gmm_anlaysis/tables/count_module.csv")

#head(count.module.df1)
count.module.df1 <- count.module.df1 %>% tidyr::separate(variable, c("Rep", "TimePoint"), "T")
#head(count.module.df1)
#hist(count.module.df1$value)

count.module.df1$Taxon <- gsub("Lachnospiraceae_bacterium", "Flavonifractor_plautii", count.module.df1$Taxon)


#tax.labs <- c("AREC","ASOE","BOVA", "BXYL","CCAT","ESIR",
#               "FRAU", "LACH","RINT", "SVAR")
unique(count.module.df1$Taxon)

count.module.df2 <-count.module.df1
#levels(count.module.df2$Taxon) <- c("AREC","ASOE","BOVA", "BXYL","CCAT","ESIR",
#               "FRAU", "FPLA","RINT", "SVAR")
mod.heat <- ggplot(count.module.df2, aes(TimePoint, Module)) +
  geom_tile(aes(fill = log(value)), color = "white", size = 0.9) + 
  theme_classic2(base_size = 12) + rotate_x_text() + 
  scale_fill_gradientn("Median abundance (log10)",
                       colors = brewer.pal(9, 'BuPu'),
                       na.value = '#f0f0f0') + facet_grid( ~ Taxon) + ylab("Metabolic Module") + xlab("Time (hr)")

# + scale_fill_gradient(high = brewer.pal(10, 'Spectral')[4], low='white', na.value = 'white')
mod.heat
ggsave("data/05_gmm/figs/module_abund_heatmap.pdf", height = 12, width = 12)

```

```{r}
strain.colors <- c(`Bacteroides ovatus` = "#1F78B4", 
                   `Bacteroides xylanisolvens` = "#B2DF8A", 
                   `Anaerobutyricum soehngenii` = "#FDBF6F", 
                   `Agathobacter rectalis`= "#33A02C", 
                   `Eubacterium siraeum` ="#FB9A99", 
                   `Faecalibacterium prausnitzii` = "#A6CEE3", 
                   `Flavonifractor plautii` = "#E31A1C", 
                   `Roseburia intestinalis` = "#CAB2D6", 
                   `Subdoligranulum variabile` = "#FF7F00", 
                   `Coprococcus catus` = "#6A3D9A")
count.module.df2$Taxon <- gsub("_", " ",count.module.df2$Taxon)

```



```{r}

#str.short.colors <- c(BOVA = "#1F78B4",BXYL = "#B2DF8A",ASOE = "#FDBF6F",   AREC= "#33A02C", ESIR ="#FB9A99",  FRAU = "#A6CEE3",  FPLA = "#E31A1C",  RINT= "#CAB2D6",  SVAR = "#FF7F00",  CCAT = "#6A3D9A")

custom_theme <- theme_bw() +
  theme(panel.grid.major = element_line(colour = "#d3d3d3"),
        panel.grid.minor = element_blank(),
        #panel.border = element_blank(), panel.background = element_blank(),
        #plot.title = element_text(size = 14, family = "Tahoma", face = "bold"),
        #text=element_text(family="Tahoma"),
        axis.title = element_text(face="bold"),
        axis.text.x=element_text(colour="black", size = 11),
        axis.text.y=element_text(colour="black", size = 9),
        axis.line = element_line(size=0.5, colour = "black"),
        strip.background = element_rect(color="black", fill="white" #, size=1.5, linetype="solid"
        )
  )

```



## Degradation  

```{r fig.width=12, fig.height=8}
#count.module.df2

module.fiber <- subset(count.module.df2, Module == "pectin degradation I" | Module == "pectin degradation II"| Module == "fructan degradation (Curated)"| Module == "arabinoxylan degradation" | Module =="starch degradation" | Module == "cellobiose degradation I (curated)")
module.fiber[module.fiber ==0]<- NA

module.fiber.plot <- ggboxplot(module.fiber, "TimePoint","value", 
                               facet.by = "Module", 
                               fill = "Taxon",
                               color = NULL,
                               palette = strain.colors, width = 0.5,
                               add= "dotplot",
                               xlab = "Time (hr)",
                               ylab = "Median module abundance",
                               scales = "free") + custom_theme +  theme(legend.text = element_text(face = "italic")) #+ theme_classic2(base_size = 12)

# + scale_fill_gradient(high = brewer.pal(10, 'Spectral')[4], low='white', na.value = 'white')
module.fiber.plot 
ggsave("data/05_gmm/figs/module_fiber_plot.pdf", height = 6, width = 12)

```


## SCFA production  
```{r fig.width=16, fig.height=12}
#count.module.df2
module.vfa <- subset(count.module.df2, Module == "propionate production IV (1,2-PD)" | Module == "propionate production I"| Module == "propionate production II"| Module == "propionate production III" | Module =="acetyl-CoA to acetate"| Module=="lactate production"| Module == "lactate consumption II" | Module=="acetyl-CoA to crotonyl-CoA" | Module == "lactate consumption III (lctABCDEF, Curated)" | Module == "pyruvate:formate lyase")
module.vfa[module.vfa ==0]<- NA

module.vfa.plot <- ggboxplot(module.vfa, "TimePoint","value", 
                             facet.by = "Module", 
                             fill = "Taxon", width = 0.5,
                             palette = strain.colors,
                             add= "dotplot",
                             xlab = "Time (hr)",
                             ylab = "Median module abundance",
                             scales = "free") + custom_theme +  theme(legend.text = element_text(face = "italic")) #+ theme_minimal(base_size = 12)


module.vfa.plot 
ggsave("data/05_gmm/figs/module_vfa_plot.pdf", height = 6, width = 12)

#ggarrange(module.fiber.plot, module.vfa.plot, 
#          ncol = 1, 
#          nrow=2, 
#          common.legend = F,
#          labels = c("a", "b"), legend = "right")
#ggsave("03_gmm_anlaysis/figs/module_fibre_vfa_plot.pdf", height = 10, width = 12)
#ggsave("03_gmm_anlaysis/figs/module.fibre.vfa.plot.tiff", height = 10, width = 12)
```


```{r fig.width=16, fig.height=12}
module.fiber.plot + module.vfa.plot + plot_layout(ncol = 1, heights = c(1, 2.5))

ggsave("data/05_gmm/figs/module.fibre.vfa.plot2.pdf", height = 10, width = 12)
```


## Mono sugars  

```{r fig.width=16, fig.height=12}
#count.module.df2

list.carbs <- c("xylose degradation", "galacturonate degradation I", "galacturonate degradation II", "lactose and galactose degradation", "maltose degradation", "arabinose degradation",	"fructose degradation",	"galactose degradation", "mannose degradation",	"rhamnose degradation")

module.carbs <- subset(count.module.df2, Module %in% list.carbs)
module.carbs[module.carbs ==0]<- NA
module.carbs.plot <- ggboxplot(module.carbs, "TimePoint","value", 
                               facet.by = "Module", 
                               fill = "Taxon",
                               palette = strain.colors, width = 0.5,
                               add= "dotplot",
                               xlab = "Time (hr)",
                               ylab = "Median module abundance",
                               scales = "free")+ custom_theme +  theme(legend.text = element_text(face = "italic"))

# + scale_fill_gradient(high = brewer.pal(10, 'Spectral')[4], low='white', na.value = 'white')
#module.carbs.plot
module.carbs.plot
ggsave("data/05_gmm/figs/module_monosac_plot.pdf", height = 6, width = 10,
       useDingbats = FALSE)

```


## Amino acid degradation  

```{r fig.width=16, fig.height=16}
aa_list <- c("alanine degradation I",
             "alanine degradation II",
             "arginine degradation I",
             "arginine degradation IV",
             "arginine degradation V",
             "aspartate degradation I",
             "aspartate degradation II",
             "cysteine degradation I",
             "cysteine degradation II",
             "glutamate degradation II",
             "glutamine degradation I",
             "glutamine degradation II",
             "glycine degradation",
             "histidine degradation",
             "isoleucine degradation",
             "leucine degradation",
             "lysine degradation I",
             "methionine degradation I",
             "serine degradation",
             "threonine degradation I",
             "threonine degradation II",
             "tryptophan degradation",
             "tyrosine degradation I")

module.aa <- subset(count.module.df2, Module %in% aa_list)
module.aa[module.aa ==0]<- NA
module.aa.plot <- ggboxplot(module.aa, "TimePoint","value", 
                            facet.by = "Module", 
                            fill = "Taxon",
                            palette = strain.colors, width = 0.5,
                            add= "dotplot",
                            xlab = "Time (hr)",
                            ylab = "Median module abundance",
                            scales = "free")+ custom_theme +  theme(legend.text = element_text(face = "italic")) 

# + scale_fill_gradient(high = brewer.pal(10, 'Spectral')[4], low='white', na.value = 'white')
module.aa.plot

ggsave("data/05_gmm/figs/moduleaminoacids_plot.pdf", 
       height = 15, width = 20,
       useDingbats = FALSE)
```


```{r fig.width=16, fig.height=16}

module.aa$taxa_ab <- toupper(abbreviate(module.aa$Taxon,method = "both.sides", minlength = 3))
module.aa.plot <- ggplot(module.aa, aes(TimePoint, Module)) +  
  geom_tile(aes(fill = log2(value)), color="white", size = 2) + 
  theme_classic2(base_size = 10) + rotate_x_text() + 
  scale_fill_gradientn("Median counts (log2)",
                       colors = brewer.pal(9, 'Greys'), 
                       na.value = '#f0f0f0') + 
  facet_grid(~ taxa_ab, scales = "free") + 
  ylab("Metabolic Module") + 
  xlab("Time (hr)") + 
  theme(axis.text.x=element_text(size=10)) 

ggsave("data/05_gmm/figs/moduleaminoacids_heatmap_plot.pdf", height = 6, width = 8)
ggsave("data/05_gmm/figs/moduleaminoacids_heatmap_plot.png", height = 6, width = 8)

```


## Amino acids biosynthesis  
```{r fig.width=16, fig.height=12}
count.module.aab <- filter(count.module.df2, grepl("biosynthesis",Module))
head(count.module.aab)
count.module.aab[count.module.aab ==0]<- NA

module.aab.plot <- ggboxplot(count.module.aab, "TimePoint","value", 
                             facet.by = "Module", 
                             fill = "Taxon",
                             palette = strain.colors, width = 0.5,
                             add= "dotplot",
                             xlab = "Time (hr)",
                             ylab = "Median module abundance",
                             scales = "free")+ custom_theme + theme(legend.text = element_text(face = "italic"))

module.aab.plot
ggsave("data/05_gmm/figs/moduleaminoacids_biosynth_plot.pdf", 
       height = 15, width = 20,
       useDingbats = FALSE)
```


```{r fig.width=16, fig.height=12}
count.module.aab$taxa_ab <- toupper(abbreviate(count.module.aab$Taxon,method = "both.sides", minlength = 3))

module.aab.plot <- ggplot(count.module.aab, aes(TimePoint, Module)) +  
  geom_tile(aes(fill = log2(value)), color="white", size = 2) + 
  theme_classic2(base_size = 10) + rotate_x_text() + 
  scale_fill_gradientn("Median counts (log2)",colors = brewer.pal(9, 'Greys'), na.value = '#f0f0f0') + 
  facet_grid(~ taxa_ab, scales = "free") + 
  ylab("Metabolic Module") + 
  xlab("Time (hr)") + 
  theme(axis.text.x=element_text(size=12))

ggsave("data/05_gmm/figs/moduleaminoacids_biosynthetsis_heatmap_plot.pdf", height = 6, width = 10)
ggsave("data/05_gmm/figs/moduleaminoacids_biosynthetsis_heatmap_plot.png", height = 6, width = 10)
```


### Lysine biosynthesis and degradation  

```{r fig.width=8, fig.height=6}
## Amino acids biosynthesis  

count.module.lys <- filter(count.module.df2, grepl("Lysine", ignore.case = TRUE,Module) )
head(count.module.lys)
unique(count.module.lys$Module)

count.module.lys[count.module.lys ==0]<- NA

count.module.lys.plot <- ggboxplot(count.module.lys, "TimePoint","value", 
                                   facet.by = "Module", 
                                   fill = "Taxon",
                                   palette = strain.colors, width = 0.5,
                                   add= "dotplot",
                                   xlab = "Time (hr)",
                                   ylab = "Median module abundance",
                                   scales = "free")+ custom_theme + theme(legend.text = element_text(face = "italic"))
count.module.lys.plot
ggsave("data/05_gmm/figs/module_lysine_plot.pdf", height = 5, width = 12)

```



## Arabinoxylan   

Below are the KOs that are considered for each of the complex substrate degradation.  

**MF0001	arabinoxylan degradation**  
K01209	K15921	K01181	K01198	K15531	K18205  

**MF0002	fructan degradation (curated, FOS)**  
K01193	K03332  
K00847  
K10117  
K10118,K10119,K10112  

**MF0003	pectin degradation I**  
K01051 
K01184,K01213	K18650  

**MF0004	pectin degradation II**  
K01051   
K18650  
K01728 K19551 K01731  
K01730  
K01815  
K00874  
K01625 K17463  
 
**MF0005	starch degradation**  
K01176 K07405 K05343   
K01200 K01208  
K00688 K16153	K00705 K01187 K15922 K01178 K01182  


```{r}

MF0001 <- c("K01209",	"K15921",	"K01181",	"K01198",	"K15531",	"K18205", 
            "K01193",	"K03332", "K00847", "K10117", "K10118","K10119","K10112", 
            "K01051", "K01184", "K01213", "K18650", 
            "K01051", "K18650", "K01728",	"K19551",	"K01731","K01730", "K01815", "K00874", "K01625",	"K17463", 
            "K01176", "K07405",	"K05343", "K01200",	"K01208", "K00688",	"K16153",	"K00705",	"K01187",	"K15922",	"K01178",	"K01182")

#brite_names <- subset(complete_table_brite, KO %in% MF0001)
KO_gene <- select(complete_table_brite, GeneName, KO)
KO_gene <- distinct(KO_gene, KO, GeneName)
head(KO_gene)
cpm_table_brite_count_name <- cpm_table_brite_count
cpm_table_brite_count_name <- merge(cpm_table_brite_count,KO_gene, by="KO")

brite_names <- subset(cpm_table_brite_count_name, KO %in% MF0001)

#brite_names
colnames(brite_names)

head(brite_gmm)

brite_names <- brite_names[,c("KO","GeneName")]

head(brite_names)

brite_names <- brite_names %>% distinct(KO, .keep_all = T)

arabinoxylan <- subset(brite_gmm, KO %in% MF0001)

deg <- merge(brite_names, arabinoxylan, by = "KO")
deg$Module <- deg$KO

Arabinoxylandegradation= c("K01209",	"K15921",	"K01181",	"K01198",	"K15531",	"K18205")
fructandegradation = c("K01193",	"K03332", "K00847", "K10117", "K10118","K10119","K10112")
pectindegradationI = c("K01051", "K01184", "K01213", "K18650" )
pectindegradationII = c("K01051", "K18650", "K01728",	"K19551",	"K01731","K01730", "K01815", "K00874", "K01625",	"K17463")
starchdegradation= c("K01176", "K07405",	"K05343", "K01200",	"K01208", "K00688",	"K16153",	"K00705",	"K01187",	"K15922",	"K01178",	"K01182")


deg <- deg %>%
  mutate(Module = case_when(KO %in% Arabinoxylandegradation ~ 'Arabinoxylan_degradation',
                            KO %in% fructandegradation ~ 'fructan_degradation',
                            KO %in% pectindegradationI ~ 'pectindegradationI',
                            KO %in% pectindegradationII~ 'pectindegradationII',
                            KO %in% starchdegradation ~ 'starch_degradation' ))

#DT::datatable(deg)
deg <- reshape2::melt(deg)
deg <- deg %>% tidyr::separate(variable, c("Rep", "TimePoint"), "T")
head(deg)

deg <- deg %>% mutate(value = replace_na(value, 0))
deg[deg ==0]<- NA 

write.csv(deg, "data/05_gmm/tables/GMM_Deg.csv")
```

```{r fig.width=16, fig.height=12}
#head(deg)

deg$Species <- gsub("Lachnospiraceae_bacterium", "Flavonifractor_plautii", deg$Species)

deg2 <- deg

deg2$Species<- gsub("_", " ", deg2$Species)

unique(deg2$TimePoint)

# create graphing function
module.graph <- function(df, na.rm = TRUE, ...){
  
  # create list of counties in data to loop over 
  mod_list <- unique(df$Module)
  
  for (i in seq_along(mod_list)) { 
    
    # create plot for each county in df 
    plot <- 
      ggboxplot(subset(df, df$Module==mod_list[i]), "TimePoint","value", 
                facet.by = "GeneName", 
                fill = "Species",
                palette = strain.colors, width = 0.5,
                add= "dotplot",
                xlab = "Time (hr)",
                ylab = "Median module abundance",
                scales = "free")+ custom_theme + theme(legend.text = element_text(face = "italic"), panel.grid = element_blank(),
                                                       panel.spacing = unit(0.3, "line"),strip.background = element_blank(),
                                                       strip.text = element_text(size = 6)) 
    
    
    # save plots as .png
    # ggsave(plot, file=paste(results, 
    #                        'projection_graphs/county_graphs/',
    #                        county_list[i], ".png", sep=''), scale=2)
    
    # save plots as .pdf
    ggsave(
      plot,
      file = paste('data/05_gmm/figs/',
                   mod_list[i], ".pdf", sep = ''),
      height = 6,
      width = 12,
      useDingbats = FALSE
    )
    
    # print plots to screen
    print(plot)
  }
  
}

#pdf("03_gmm_anlaysis/figs/module_kos_carbs_plot.pdf", height = 6, width = 12)
module.graph(deg2)
#dev.off()


mod_list <- unique(deg2$Module)
de3 <- subset(deg2, Module=="pectin degradationI" && Module=="pectin degradationII")
pect1 <- 
  ggboxplot(subset(deg2, Module=="pectindegradationI" | Module=="pectindegradationII"), "TimePoint","value", 
            facet.by = "GeneName", 
            fill = "Species",
            palette = strain.colors, width = 0.5,
            add= "dotplot",
            xlab = "Time (hr)",
            ylab = "Median module abundance",
            scales = "free")+ 
  custom_theme + 
  theme(legend.text = element_text(face = "italic"), 
        panel.grid = element_blank(),
        panel.spacing = unit(0.3, "line"),
        strip.background = element_blank(),
        strip.text = element_text(size = 6)) 

ggsave("data/05_gmm/figs/pectindegradationI_n_II.pdf",
       height = 4,
       width = 8,
       useDingbats = FALSE
)

```


#### Lysine degration KOs
```{r fig.width=16, fig.height=12}

lysine_KO <- c("K01843","K01844","K18011","K18012","K18013", "K18014")


brite_names2 <- subset(complete_table_brite, KO %in% lysine_KO)
#brite_names
colnames(brite_names2)

head(brite_gmm)

brite_names2 <- brite_names2[,c("KO","GeneName")]

head(brite_names2)

brite_names2 <- brite_names2 %>% distinct(KO, .keep_all = T)

lysineD <- subset(brite_gmm, KO %in% lysine_KO)

Lysdeg <- merge(brite_names2, lysineD, by = "KO")
Lysdeg$Module <- Lysdeg$KO

#DT::datatable(deg)
Lysdeg <- reshape2::melt(Lysdeg)
Lysdeg <- Lysdeg %>% tidyr::separate(variable, c("Rep", "TimePoint"), "T")
head(Lysdeg)

Lysdeg <- Lysdeg %>% mutate(value = replace_na(value, 0))
Lysdeg[Lysdeg ==0]<- NA 


head(Lysdeg)
#Lysdeg$Species <- Lysdeg$BacterialStrain
Lysdeg$Species <- gsub("Lachnospiraceae_bacterium", "Flavonifractor_plautii",Lysdeg$Species)

Lysdeg$Species <- gsub("_"," ",Lysdeg$Species)

lysplot <- 
  ggboxplot(Lysdeg, "TimePoint","value", 
            facet.by = "GeneName", 
            fill = "Species",
            palette = strain.colors, width = 0.5,
            add= "dotplot",
            xlab = "Time (hr)",
            ylab = "Median module abundance",
            scales = "free")+ custom_theme + 
  theme(legend.text = element_text(face = "italic"), panel.grid = element_blank(),
        panel.spacing = unit(0.3, "line"),strip.background = element_blank(),
        strip.text = element_text(size = 8)) 

ggsave("data/05_gmm/figs/lysine_deg_KO.pdf",
       height = 7,
       width = 14,
       useDingbats = FALSE)

```


```{r eval=FALSE}
sessionInfo()
```

