---
title: "Assembly diet based minimal microbiome (Db-MM10)"
subtitle: "Link locus tags to KEGG KO IDs for pathway and metabolic module analysis"
author: "Sudarshan A. Shetty"
date: "`r date()`"
output:
  workflowr::wflow_html:
    toc: yes
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---


# Introduction  

For detailed metabolic pathway analysis, we focus on KEGG as a useful resource. We will link locus tags to KEGG KO IDs for pathway and metabolic module analysis.   

This way we can, for each sample, reconstruct metabolic pathways that are active in the DbMM.  

## Setup  
```{r, message=FALSE, warning=FALSE}

# import other necessary packages
suppressPackageStartupMessages({
  library(DESeq2)
  library(tidyverse)
  #library(EnhancedVolcano)
  library(ggplot2)
  library(data.table)
  library(qgraph)
})

#library(optparse)
```

For this we need to first link the locus tags with respective KO ids that we identified by doing GhostKola for concatenated amino acid sequences that was used as database for DIAMOND.  

Link the locus tags to KEGG orthologs.  

# Read KEEG BRITE classification  
The KEGG htext was downloaded from KEGG website.  

```{r}

keg.brite <- fread("data_raw/04_metatrans/kegg_brite/KEGG_htext_classificationDb.txt", header = F, sep = "\t")

#head(keg.brite)
#DT::datatable(keg.brite.x)
colnames(keg.brite) <- c("Level_1", "Level_2", "Level_3", "Level_4", "KO")

```


Edit to seprate last column.  

```{r}

keg.brite2 <- keg.brite %>% mutate(Level_4 = str_replace(Level_4, "\\s", "|")) %>%
  separate(Level_4, into = c("KOID", "GeneName"), sep = "\\|")

#head(keg.brite2)

#write.csv(keg.brite2, "tables/01_kegg_brite_full_list.csv")
  
#DT::datatable(keg.brite2)
```


# GhostKola of Db-MM  
Load the KO annotation for DB-MM locus tags obtained form online GHOSTKola annotation. The concatenated amino acid fasta of all then strains was uploaded to GhostKola for annotation using default parameters. 

```{r}

ghostko <- fread("data_raw/04_metatrans/kegg_brite/GhostKola_DbMM_ko_list.txt", header = F, sep = "\t")
#head(ghostko)
colnames(ghostko) <- c("LocusTag", "KO")
#head(ghostko)
```


# Reviewers  
Check for KO similarity between DbMM genomes.  
##General all KOs
```{r}
ghostko.df <- as.data.frame(ghostko) 

ghostko.df$LocusTag <- stringr::str_extract(ghostko.df$LocusTag, "[^_]*_[^_]*")
ghostko.df <- ghostko.df %>% 
  filter(!is.na(KO)) %>% 
  group_by(LocusTag) %>% 
  distinct(KO) %>% 
  mutate(presence.absence = ifelse(is.na(KO), 0, 1)) %>% 
  pivot_wider(id_cols = KO , names_from = LocusTag, values_from = presence.absence) %>% 
  mutate(across(everything(), .fns = ~replace_na(.,0))) %>% 
  column_to_rownames("KO")

head(ghostko.df)
#t(ghostko.df)
# check for Genome KO similarity
#dmm.dist <- vegan::vegdist(t(ghostko.df), method="jaccard") 
#hist(dmm.dist, x="Jaccard distance")
#head(dmm.dist)
dist4 <- 1-dist(t(ghostko.df), method = "binary")
dist4DF <- dist4 %>% as.matrix() %>%  
  as.data.frame() %>% 
  reshape2::melt() %>% 
  group_by(variable) %>% 
  filter(value !=0 & !variable %in% c("Bacteroides_ovatus", "Bacteroides_xylanisolvens")) %>% 
  summarise(mean = mean(value),
            median = median(value),
            sd = sd(value)) %>% 
  arrange(median)
```

### Plot all ko dist  

```{r}
qgraph(dist4, layout='spring', vsize=3, 
       edge.labels=TRUE, labels= rownames(as.matrix(dist4)),
       label.cex = 10,
       edge.color = "brown3")
```



## Subset of KOs
```{r}
ghostko.df <- as.data.frame(ghostko) 
ghostko.df$LocusTag <- stringr::str_extract(ghostko.df$LocusTag, "[^_]*_[^_]*")

keg.brite.dmm <- subset(keg.brite2, KO %in% unique(ghostko.df$KO))
#ghostko.df$KO <- rownames(ghostko.df)
ghostko.df.brite <- merge(ghostko.df, keg.brite.dmm, by = "KO")
unique(ghostko.df.brite$Level_2)
sel <- c("09105 Amino acid metabolism", "09101 Carbohydrate metabolism", "09102 Energy metabolism",
"09106 Metabolism of other amino acids","09141 Transport and catabolism","09106 Metabolism of other amino acids","09191 Unclassified: metabolism","09108 Metabolism of cofactors and vitamins","09107 Glycan biosynthesis and metabolism","09131 Membrane transport")
ghostko.df.brite <- ghostko.df.brite %>% 
  filter(Level_2 %in% sel)
head(ghostko.df.brite)

ghostko.df.brite <- select(ghostko.df.brite, c("KO", "LocusTag"))

ghostko.df.brite$LocusTag <- stringr::str_extract(ghostko.df.brite$LocusTag, "[^_]*_[^_]*")
ghostko.df.brite <- ghostko.df.brite %>% 
  filter(!is.na(KO)) %>% 
  group_by(LocusTag) %>% 
  distinct(KO) %>% 
  mutate(presence.absence = ifelse(is.na(KO), 0, 1)) %>% 
  pivot_wider(id_cols = KO , names_from = LocusTag, values_from = presence.absence) %>% 
  mutate(across(everything(), .fns = ~replace_na(.,0))) %>% 
  column_to_rownames("KO")

dist4.brite <- 1-dist(t(ghostko.df.brite), method = "binary")
dist4DF.brite <- dist4.brite %>% as.matrix() %>%  
  as.data.frame() %>% 
  reshape2::melt() %>% 
  group_by(variable) %>% 
  filter(value !=0 & !variable %in% c("Bacteroides_ovatus", "Bacteroides_xylanisolvens")) %>% 
  summarise(mean = mean(value),
            median = median(value),
            sd = sd(value))

```

### Plot subset ko dist   
```{r}
qgraph(dist4.brite, layout='spring', vsize=3, 
       edge.labels=TRUE, labels= rownames(as.matrix(dist4.brite)),
       label.cex = 10,
       edge.color = "brown3")
```

Now read in this session, the `raw_counts_tables.rds` file created from `01_meta_counts.rmd` file.  

```{r}

# Read the raw counts table 
complete_table <- readRDS("data/04_metatrans/rds/raw_counts_tables.rds")
# filter to keep only those that are annotated in Db-MM
ko_locus <- subset(ghostko, ghostko$LocusTag %in% rownames(complete_table))

dim(ko_locus)
dim(complete_table)

complete_table.ko <- complete_table
complete_table.ko$LocusTag <- rownames(complete_table)
complete_table.ko <- merge(complete_table.ko, ko_locus, by = "LocusTag")
#DT::datatable(complete_table.ko)

#dim(complete_table.ko)
saveRDS(complete_table.ko, "data/04_metatrans/rds/complete_table_with_ko.rds")
write.table(complete_table.ko, "data/04_metatrans/tables/complete_table_with_ko.txt", sep="\t", col.names = F)

```

Now add BRITE classification.

```{r}

# keg.brite2
keg.brite.sub <- subset(keg.brite2, KO %in% complete_table.ko$KO)

#dim(keg.brite2)
#dim(keg.brite.sub)
#dim(ghostko)
#dim(complete_table.ko)

keg.brite.sub.na <- subset(keg.brite.sub, !(is.na(KO)))
dim(keg.brite.sub.na)

complete_table.brite <- merge(complete_table.ko, keg.brite.sub, by = "KO")

# Store all intermediate files to check and explore manually interesting features 
write.csv(keg.brite.sub.na, "data/04_metatrans/tables/kegg_brite_sub_without_NA.csv")
dim(keg.brite.sub.na)
dim(complete_table.brite)
write.csv(complete_table.brite, "data/04_metatrans/tables/complete_table_ko_brite.csv")

```

KEGG KO-metabolite mapping  
```{r}
head(complete_table.brite)
```


## Prep file for input to gmm species-function anlaysis   

We want to make use of the manually curated database of human gut metabolic modules by [Vieira-Silva et al. 2016](https://www.nature.com/articles/nmicrobiol201688).  

We need to format the KO counts file.

```{r}

complete_table.brite$BacterialStrain <- complete_table.brite$LocusTag

complete_table.brite <-separate(complete_table.brite, BacterialStrain, c("Bacteria", "Locus"), extra = "merge", fill = "right")
complete_table.brite <-separate(complete_table.brite, Locus, c("Species", "Locus"), extra = "merge", fill = "right")
complete_table.brite <- complete_table.brite %>% unite(BacterialStrain, Bacteria,Species,Species) 

head(complete_table.brite)

complete_table.brite <- complete_table.brite %>% distinct(LocusTag, .keep_all = TRUE)

head(complete_table.brite)
unique(complete_table.brite$BacterialStrain)
class(complete_table.brite)

# store this also for manual inspection
write.csv(complete_table.brite, "data/04_metatrans/tables/complete_table_brite_species.csv")

```


### Save file

```{r}

complete_table.brite_filter <- complete_table.brite[, c(-15)]

head(complete_table.brite_filter)
saveRDS(complete_table.brite_filter, "data/04_metatrans/rds/complete_table_brite_filter.rds")

```


Next, we will prepare the files for KEGG mapping using `pathview R pkg`. This can take fold changes of KOs and mapping to KEGG maps.  
This is at community level. Hence, we aggregate KO counts per sample.  
## Aggregate by KEGG IDs  

```{r}

count.aggregate.kegg <- complete_table.brite %>% group_by(KO) %>% 
  summarise(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#colSums(count.aggregate.kegg)

head(count.aggregate.kegg)

colnames(count.aggregate.kegg) <- c("KOID","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
# store this
write.table(count.aggregate.kegg, "data/04_metatrans/tables/count_aggregate_kegg.txt", sep="\t", col.names = T, row.names = F)

head(count.aggregate.kegg)
```

## DESeq2 analysis of KOs  

```{r}
count.aggregate.kegg <- as.data.frame(count.aggregate.kegg)
rownames(count.aggregate.kegg) <- count.aggregate.kegg$KOID
count.aggregate.kegg <- count.aggregate.kegg[,-1]
head(count.aggregate.kegg)
#DT::datatable(count.aggregate.kegg)

coldata.df <- data.frame(SampleID = c("B1T24","B2T24","B3T24","B1T48","B2T48","B3T48"),
                            condition = c("24h","24h","24h", "48h","48h","48h"))
coldata.df
#write.table(coldata.df, "tables/coldata_df.txt", sep = "\t")
dds <- DESeqDataSetFromMatrix(countData = count.aggregate.kegg,
                              colData = coldata.df,
                              design = ~ condition)
dds

```

Filter our KOs with low counts across samples.  
```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

dds <- DESeq(dds,fitType= "local")

res <- results(dds)
ko_results <- data.frame(res)
sorted_ko_results <- ko_results[order(-ko_results$baseMean),]
colnames(sorted_ko_results)[1] <- "LocusTag"
head(sorted_ko_results)
saveRDS(sorted_ko_results, "data/04_metatrans/rds/sorted_ko_deseq2_results.rds")

```

The `sorted_ko_deseq2_results.rds` will be used for `kegg_pathway_analysis`.  

```{r eval=FALSE}
sessionInfo()

```


