---
title: "Assembly diet based minimal microbiome (Db-MM10)"
subtitle: "Reverse ecology based microbial interactions using metatranscriptome"
author: "Sudarshan A. Shetty"
date: "`r date()`"
output:
  workflowr::wflow_html:
    toc: yes
    toc_depth: 2
    toc_float: true
editor_options: 
  chunk_output_type: console
---


# Introduction  
Following the approach using [RevEcoR](https://cran.r-project.org/web/packages/RevEcoR/vignettes/RevEcoR.html). The reverse ecology framework is used for reconstruction of metabolic networks and idenditifying the set of compounds which is used for predicting interactions between the species as described by [Borenstein E, Kupiec M, Feldman M W, et al. Large-scale reconstruction and phylogenetic analysis of metabolic environments. Proceedings of the National Academy of Sciences, 2008, 105(38): 14482-14487.](https://www.pnas.org/content/105/38/14482.long)   

Note: Requires Internet connection.  

## Set up 

```{r}

suppressPackageStartupMessages({
  library(RevEcoR)
  library(tidyverse)
  library(reshape2)
  library(qgraph)
  library(ggplot2)
  #library(patchwork)
  library(ggpubr)
  library(RColorBrewer)
  #library(tidyverse)
})
```

Read the table with all KOs and species locus tags `complete_table_brite_filter.rds` located in the folder `data/04_metatrans/rds`

```{r}

complete_table.brite_filter <- readRDS("data/04_metatrans/rds/complete_table_brite_filter.rds")

#DT::datatable(complete_table.brite_filter)

complete_table.brite_filter.uniq <- complete_table.brite_filter %>% distinct(LocusTag, .keep_all = TRUE)
head(complete_table.brite_filter.uniq)

dim(complete_table.brite_filter.uniq)
colnames(complete_table.brite_filter.uniq)

complete_table.brite_filter <- complete_table.brite_filter.uniq
```

```{r}
#############################################################################################
arec <- subset(complete_table.brite_filter, BacterialStrain == "Agathobacter_rectalis")
# write.csv(arec, "arec_complete_table.csv")
#arec <- arec[,-9]
arec <- arec %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

head(arec)
#head(df.gene_table.x)

colnames(arec) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")

#############################################################################################
asoe <- subset(complete_table.brite_filter, BacterialStrain == "Anaerobutyricum_soehngenii")
#asoe <- asoe[,-9]
asoe <- asoe %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(asoe) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
head(asoe)
#############################################################################################

bova <- subset(complete_table.brite_filter, BacterialStrain == "Bacteroides_ovatus")
#bova <- bova[,-9]
bova <- bova %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(bova) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")

#############################################################################################


bxyl <- subset(complete_table.brite_filter, BacterialStrain == "Bacteroides_xylanisolvens")
#bxyl <- bxyl[,-9]
bxyl <- bxyl %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(bxyl) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
#############################################################################################

ccat <- subset(complete_table.brite_filter, BacterialStrain == "Coprococcus_catus")
#ccat <- ccat[,-9]
ccat <- ccat %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(ccat) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")

#############################################################################################

esir <- subset(complete_table.brite_filter, BacterialStrain == "Eubacterium_siraeum")
#esir <- esir[,-9]
esir <- esir %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(esir) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
#############################################################################################
fpra <- subset(complete_table.brite_filter, BacterialStrain == "Faecalibacterium_prausnitzii")
#fpra <- fpra[,-9]
fpra <- fpra %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(fpra) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
#############################################################################################

lach <- subset(complete_table.brite_filter, BacterialStrain == "Lachnospiraceae_bacterium")
#lach <- lach[,-9]
lach <- lach %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(lach) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
#############################################################################################


rint <- subset(complete_table.brite_filter, BacterialStrain == "Roseburia_intestinalis")
#rint <- rint[,-9]
rint <- rint %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(rint) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")

#############################################################################################
svar <- subset(complete_table.brite_filter, BacterialStrain == "Subdoligranulum_variabile")
#svar <- svar[,-9]
svar <- svar %>% group_by(KOID, GeneName) %>% 
  summarize(sum(B1T24), sum(B2T24), sum(B3T24),sum(B1T48), sum(B2T48), sum(B3T48))

#head(df.gene_table.x)
colnames(svar) <- c("KO ID","Definition","B1T24","B2T24","B3T24","B1T48","B2T48","B3T48")
head(svar)

```

```{r}
head(bova)
bova.s <- as.data.frame(bova)
rownames(bova.s) <- bova.s$`KO ID`
#which(bova.s[rowSums(bova.s[,3:8]) >10,])
#bxyl
bxyl.s <- as.data.frame(bxyl)
rownames(bxyl.s) <- bxyl.s$`KO ID`

arec.s <- as.data.frame(arec)
rownames(arec.s) <- arec.s$`KO ID`

asoe.s <- as.data.frame(asoe)
rownames(asoe.s) <- asoe.s$`KO ID`

ccat.s <- as.data.frame(ccat)
rownames(ccat.s) <- ccat.s$`KO ID`

esir.s <- as.data.frame(esir)
rownames(esir.s) <- esir.s$`KO ID`

fpra.s <- as.data.frame(fpra)
rownames(fpra.s) <- fpra.s$`KO ID`

lach.s <- as.data.frame(lach)
rownames(lach.s) <- lach.s$`KO ID`


rint.s <- as.data.frame(rint)
rownames(rint.s) <- rint.s$`KO ID`

svar.s <- as.data.frame(svar)
rownames(svar.s) <- svar.s$`KO ID`

```


## Reconstruct metabolic network for each bacterial strain using transcript data  

When running the codes for first time change the `eval=FALSE` to `TRUE` 
```{r eval=T}
Bacteroides_ovatu_3_8_47FAA.net <- reconstructGsMN(bova.s, RefData = RefDbcache) 
seed.set.bova <- getSeedSets(Bacteroides_ovatu_3_8_47FAA.net)

Bacteroides_sp_2_1_22.net <- reconstructGsMN(bxyl.s, RefData = RefDbcache) 

#Bacteroides_sp_3_1_23.net <- reconstructGsMN(Bacteroides_sp_3_1_23, RefData = RefDbcache)


Coprococcus_catus.net <- reconstructGsMN(ccat.s, RefData = RefDbcache)


Eubacterium_hallii_L2_7.net <- reconstructGsMN(asoe.s, RefData = RefDbcache)


Eubacterium_rectale_DSM_17629.net <- reconstructGsMN(arec.s, RefData = RefDbcache)


Faecalibacterium_prausnitzii_A2_165.net  <- reconstructGsMN(fpra.s, 
                                                            RefData = RefDbcache)


Lachnospiraceae_bacterium_7_1_58FAA.net <- reconstructGsMN(lach.s,
                                                           RefData = RefDbcache) 

Roseburia_intestinalis_L1_82.net <- reconstructGsMN(rint.s, 
                                                    RefData = RefDbcache)

#Ruminococcus_bromii_l2.net <- reconstructGsMN(Ruminococcus_bromii_l2, 
#                                    RefData = RefDbcache) 


Subdoligranulum_variabile_DSM_15176.net <- reconstructGsMN(svar.s,
                                                           RefData = RefDbcache) 

Eubacterium_siraeum_DSM_15702.net <- reconstructGsMN(esir.s,
                                                     RefData = RefDbcache)

```


# Calculate interaction indices   

When running the codes for first time change the `eval=FALSE` to `TRUE` 
```{r eval=T}

cooperation.index<-calculateCooperationIndex(Bacteroides_ovatu_3_8_47FAA.net,
                                             Bacteroides_sp_2_1_22.net,
                                             Coprococcus_catus.net,
                                             Eubacterium_hallii_L2_7.net,
                                             Eubacterium_rectale_DSM_17629.net,
                                             Faecalibacterium_prausnitzii_A2_165.net,
                                             Lachnospiraceae_bacterium_7_1_58FAA.net,
                                             Roseburia_intestinalis_L1_82.net,
                                             Subdoligranulum_variabile_DSM_15176.net,
                                             Eubacterium_siraeum_DSM_15702.net)

saveRDS(cooperation.index, "data/04_metatrans/rds/RevEco_cooperation.index.rds")

```

Read in the stored cooperative indices  
```{r}

cooperation.index <- readRDS("data/04_metatrans/rds/RevEco_cooperation.index.rds")


complementarity.index_2018 <- cooperation.index$complementarity.index

competition.index_2018 <- cooperation.index$competition.index

write.csv(complementarity.index_2018, "data/04_metatrans/tables/DbMM10_complementarity_index.csv") 

write.csv(competition.index_2018, "data/04_metatrans/tables/DbMM10_competition_index.csv") 
```


add names to competition matrix  

```{r}
rownames(competition.index_2018) <- c("Bacteroides_ovatus",
                                      "Bacteroides_xylanisolvens",
                                      "Coprococcus_catus",
                                      "Anaerobutyricum_soehngenii",
                                      "Agathobacter_rectalis",
                                      "Faecalibacterium_prausnitzii",
                                      "Flavonifractor_plautii",
                                      "Roseburia_intestinalis",
                                      "Subdoligranulum_variabile",
                                      "Eubacterium_siraeum")

colnames(competition.index_2018) <- c("Bacteroides_ovatus",
                                      "Bacteroides_xylanisolvens",
                                      "Coprococcus_catus",
                                      "Anaerobutyricum_soehngenii",
                                      "Agathobacter_rectalis",
                                      "Faecalibacterium_prausnitzii",
                                      "Flavonifractor_plautii",
                                      "Roseburia_intestinalis",
                                      "Subdoligranulum_variabile",
                                      "Eubacterium_siraeum")
```

add names to complementatrity matrix
```{r}

rownames(complementarity.index_2018) <- c("Bacteroides_ovatus",
                                          "Bacteroides_xylanisolvens",
                                          "Coprococcus_catus",
                                          "Anaerobutyricum_soehngenii",
                                          "Agathobacter_rectalis",
                                          "Faecalibacterium_prausnitzii",
                                          "Flavonifractor_plautii",
                                          "Roseburia_intestinalis",
                                          "Subdoligranulum_variabile",
                                          "Eubacterium_siraeum")


colnames(complementarity.index_2018) <- c("Bacteroides_ovatus",
                                          "Bacteroides_xylanisolvens",
                                          "Coprococcus_catus",
                                          "Anaerobutyricum_soehngenii",
                                          "Agathobacter_rectalis",
                                          "Faecalibacterium_prausnitzii",
                                          "Flavonifractor_plautii",
                                          "Roseburia_intestinalis",
                                          "Subdoligranulum_variabile",
                                          "Eubacterium_siraeum")
```

```{r}
# set colors  
strain.colors <- c(Bacteroides_ovatus = "#1F78B4", 
                   Bacteroides_xylanisolvens = "#B2DF8A", 
                   Anaerobutyricum_soehngenii = "#FDBF6F", 
                   Agathobacter_rectalis= "#33A02C", 
                   Eubacterium_siraeum ="#FB9A99", 
                   Faecalibacterium_prausnitzii = "#A6CEE3", 
                   Flavonifractor_plautii = "#E31A1C", 
                   Roseburia_intestinalis= "#CAB2D6", 
                   Subdoligranulum_variabile = "#FF7F00", 
                   Coprococcus_catus = "#6A3D9A")

```

# Plot interaction indices 
# Transcriptomics


```{r}

colnames(complementarity.index_2018) <- gsub("_", " ", colnames(complementarity.index_2018))
rownames(complementarity.index_2018) <- gsub("_", " ", rownames(complementarity.index_2018))

nums.a1 <- round(complementarity.index_2018, 3)
pheatmap::pheatmap(as.matrix(complementarity.index_2018),
                   display_numbers =nums.a1,
                   filename = "data/04_metatrans/figs/R1_TranscriptComplementarity.pdf",
                   height = 6, width = 8,
                   main = "Transcriptome Complementarity")
  
colnames(competition.index_2018) <- gsub("_", " ", colnames(competition.index_2018))
rownames(competition.index_2018) <- gsub("_", " ", rownames(competition.index_2018))


nums.b1 <- round(competition.index_2018, 3)
pheatmap::pheatmap(as.matrix(competition.index_2018),
                   display_numbers =nums.b1,
                   filename = "data/04_metatrans/figs/R1_TranscriptCompetition.pdf",
                   height = 6, width = 8,
                   main = "Transcriptome Competition")
```


# Genomic  

## Reviewer-1  

Reconstruct metabolic network for each bacterial strain using Genomics data  

When running the codes for first time change the `eval=FALSE` to `TRUE`  

```{r}
library(data.table)
# We need rownames and KO ID
ghostko <- fread("data_raw/04_metatrans/kegg_brite/GhostKola_DbMM_ko_list.txt", header = F, sep = "\t")
#head(ghostko)
colnames(ghostko) <- c("LocusTag", "KO")
ghostko.df <- as.data.frame(ghostko) 

ghostko.df$LocusTag <- stringr::str_extract(ghostko.df$LocusTag, "[^_]*_[^_]*")
ghostko.df <- ghostko.df %>% 
  filter(!is.na(KO)) %>% 
  group_by(LocusTag) %>% 
  distinct(KO) %>% 
  mutate(presence.absence = ifelse(is.na(KO), 0, 1)) %>% 
  pivot_wider(id_cols = KO , names_from = LocusTag, values_from = presence.absence) %>% 
  mutate(across(everything(), .fns = ~replace_na(.,0))) %>% 
  column_to_rownames("KO")
```

```{r eval=T}

genome_ko_df <- NULL
subDF <- NULL
for(i in colnames(ghostko.df)){
  
  subDF <-  as.data.frame(ghostko.df[,i])
  colnames(subDF)[1] <- "binary"
  rownames(subDF) <- rownames(ghostko.df)
  subDF$`KO ID` <- rownames(subDF)
  subDF <- subset(subDF, binary == 1)
  subDF <- select(subDF, c("KO ID"))
  genome_ko_df[[i]] <- subDF
  
}


gBacteroides_ovatu_3_8_47FAA.net <- reconstructGsMN(genome_ko_df$Bacteroides_ovatus, RefData = RefDbcache) 


gBacteroides_sp_2_1_22.net <- reconstructGsMN(genome_ko_df$Bacteroides_xylanisolvens, RefData = RefDbcache) 

#Bacteroides_sp_3_1_23.net <- reconstructGsMN(Bacteroides_sp_3_1_23, RefData = RefDbcache)


gCoprococcus_catus.net <- reconstructGsMN(genome_ko_df$Coprococcus_catus, RefData = RefDbcache)


gEubacterium_hallii_L2_7.net <- reconstructGsMN(genome_ko_df$Anaerobutyricum_soehngenii, RefData = RefDbcache)


gEubacterium_rectale_DSM_17629.net <- reconstructGsMN(genome_ko_df$Agathobacter_rectalis, RefData = RefDbcache)


gFaecalibacterium_prausnitzii_A2_165.net  <- reconstructGsMN(genome_ko_df$Faecalibacterium_prausnitzii, 
                                                             RefData = RefDbcache)


gLachnospiraceae_bacterium_7_1_58FAA.net <- reconstructGsMN(genome_ko_df$Lachnospiraceae_bacterium,
                                                            RefData = RefDbcache) 

gRoseburia_intestinalis_L1_82.net <- reconstructGsMN(genome_ko_df$Roseburia_intestinalis, 
                                                     RefData = RefDbcache)

gSubdoligranulum_variabile_DSM_15176.net <- reconstructGsMN(genome_ko_df$Subdoligranulum_variabile,
                                                            RefData = RefDbcache) 

gEubacterium_siraeum_DSM_15702.net <- reconstructGsMN(genome_ko_df$Eubacterium_siraeum,
                                                      RefData = RefDbcache)

```


# Calculate interaction indices   

When running the codes for first time change the `eval=FALSE` to `TRUE` 
```{r eval=T}

gcooperation.index <- calculateCooperationIndex(gBacteroides_ovatu_3_8_47FAA.net,
                                                gBacteroides_sp_2_1_22.net,
                                                gCoprococcus_catus.net,
                                                gEubacterium_hallii_L2_7.net,
                                                gEubacterium_rectale_DSM_17629.net,
                                                gFaecalibacterium_prausnitzii_A2_165.net,
                                                gLachnospiraceae_bacterium_7_1_58FAA.net,
                                                gRoseburia_intestinalis_L1_82.net,
                                                gSubdoligranulum_variabile_DSM_15176.net,
                                                gEubacterium_siraeum_DSM_15702.net)

saveRDS(gcooperation.index, "data/04_metatrans/rds/GenomicRevEco_cooperation.index.rds")


```

Read in the stored cooperative indices  
```{r}

gcooperation.index <- readRDS("data/04_metatrans/rds/GenomicRevEco_cooperation.index.rds")


gcomplementarity.index_2018 <- gcooperation.index$complementarity.index

gcompetition.index_2018 <- gcooperation.index$competition.index

write.csv(gcomplementarity.index_2018, "data/04_metatrans/tables/GenomicDbMM10_complementarity_index.csv") 

write.csv(gcompetition.index_2018, "data/04_metatrans/tables/GenomicDbMM10_competition_index.csv") 
```


add names to competition matrix  

```{r}
rownames(gcompetition.index_2018) <- c("Bacteroides_ovatus",
                                      "Bacteroides_xylanisolvens",
                                      "Coprococcus_catus",
                                      "Anaerobutyricum_soehngenii",
                                      "Agathobacter_rectalis",
                                      "Faecalibacterium_prausnitzii",
                                      "Flavonifractor_plautii",
                                      "Roseburia_intestinalis",
                                      "Subdoligranulum_variabile",
                                      "Eubacterium_siraeum")

colnames(gcompetition.index_2018) <- c("Bacteroides_ovatus",
                                      "Bacteroides_xylanisolvens",
                                      "Coprococcus_catus",
                                      "Anaerobutyricum_soehngenii",
                                      "Agathobacter_rectalis",
                                      "Faecalibacterium_prausnitzii",
                                      "Flavonifractor_plautii",
                                      "Roseburia_intestinalis",
                                      "Subdoligranulum_variabile",
                                      "Eubacterium_siraeum")
```

add names to complementatrity matrix
```{r}

rownames(gcomplementarity.index_2018) <- c("Bacteroides_ovatus",
                                          "Bacteroides_xylanisolvens",
                                          "Coprococcus_catus",
                                          "Anaerobutyricum_soehngenii",
                                          "Agathobacter_rectalis",
                                          "Faecalibacterium_prausnitzii",
                                          "Flavonifractor_plautii",
                                          "Roseburia_intestinalis",
                                          "Subdoligranulum_variabile",
                                          "Eubacterium_siraeum")


colnames(gcomplementarity.index_2018) <- c("Bacteroides_ovatus",
                                          "Bacteroides_xylanisolvens",
                                          "Coprococcus_catus",
                                          "Anaerobutyricum_soehngenii",
                                          "Agathobacter_rectalis",
                                          "Faecalibacterium_prausnitzii",
                                          "Flavonifractor_plautii",
                                          "Roseburia_intestinalis",
                                          "Subdoligranulum_variabile",
                                          "Eubacterium_siraeum")
```


## Genomics
```{r}

colnames(gcomplementarity.index_2018) <- gsub("_", " ", colnames(gcomplementarity.index_2018))
rownames(gcomplementarity.index_2018) <- gsub("_", " ", rownames(gcomplementarity.index_2018))

nums.a <- round(gcomplementarity.index_2018, 3)
pheatmap::pheatmap(as.matrix(gcomplementarity.index_2018),
                   display_numbers =nums.a,
                   filename = "data/04_metatrans/figs/R1_GenomicComplementarity.pdf",
                   height = 6, width = 8,
                   main = "Genome Complementarity")

colnames(gcompetition.index_2018) <- gsub("_", " ", colnames(gcompetition.index_2018))
rownames(gcompetition.index_2018) <- gsub("_", " ", rownames(gcompetition.index_2018))
nums.b <- round(gcompetition.index_2018, 3)
pheatmap::pheatmap(as.matrix(gcompetition.index_2018),
                   display_numbers =nums.b,
                   filename = "data/04_metatrans/figs/R1_GenomicCompetition.pdf",
                   height = 6, width = 8,
                  main = "Genome Competition")
```




## Reviewer-1  

Reconstruct metabolic network for each bacterial strain using Genomics data  

When running the codes for first time change the `eval=FALSE` to `TRUE`  


### Caculate what percent of genome KO are used in transcriptome KOs  

```{r}
# Transcriptomics
head(bova)
b.ova.s.chk <- length(intersect(rownames(genome_ko_df$Bacteroides_ovatus),
                                 bova.s[,1]))/length(rownames(genome_ko_df$Bacteroides_ovatus)) 

b.xyl.chk <- length(intersect(rownames(genome_ko_df$Bacteroides_ovatus), 
                              bxyl.s [,1]))/length(rownames(genome_ko_df$Bacteroides_xylanisolvens))

arec.s.chk <- length(intersect(rownames(genome_ko_df$Agathobacter_rectalis), 
                               arec.s[,1]))/length(rownames(genome_ko_df$Agathobacter_rectalis))

asoe.s.chk <- length(intersect(rownames(genome_ko_df$Anaerobutyricum_soehngenii), 
                               asoe.s [,1]))/length(rownames(genome_ko_df$Anaerobutyricum_soehngenii))

ccat.s.chk <- length(intersect(rownames(genome_ko_df$Coprococcus_catus), 
                               ccat.s [,1]))/length(rownames(genome_ko_df$Coprococcus_catus))

esir.s.chk <- length(intersect(rownames(genome_ko_df$Eubacterium_siraeum), 
                               esir.s [,1]))/length(rownames(genome_ko_df$Eubacterium_siraeum))

fpra.s.chk <- length(intersect(rownames(genome_ko_df$Faecalibacterium_prausnitzii), 
                               fpra.s [,1]))/length(rownames(genome_ko_df$Faecalibacterium_prausnitzii))

lach.s.chk <- length(intersect(rownames(genome_ko_df$Lachnospiraceae_bacterium), 
                               lach.s [,1]))/length(rownames(genome_ko_df$Lachnospiraceae_bacterium))

rint.s.chk <- length(intersect(rownames(genome_ko_df$Roseburia_intestinalis), 
                               rint.s [,1]))/length(rownames(genome_ko_df$Roseburia_intestinalis))

svar.s.chk <- length(intersect(rownames(genome_ko_df$Subdoligranulum_variabile), 
                               esir.s [,1]))/length(rownames(genome_ko_df$Subdoligranulum_variabile))

pct_genome_expressed <- tibble(names= c("BOVA", "BXYL", "AREC","ASOE","CCAT",
                                        "ESIR","FPRA","FPLA","RINT","SVAR"),
                               pct_exps = c(b.ova.s.chk,b.xyl.chk,arec.s.chk,
                                            asoe.s.chk, ccat.s.chk,esir.s.chk,
                                            fpra.s.chk,lach.s.chk,rint.s.chk,
                                            svar.s.chk))
mean(pct_genome_expressed$pct_exps)
sd(pct_genome_expressed$pct_exps)
```
0.9472932
0.09726227

## Reve 

How much of human gut metagenome KO is coverred by Db-MM

```{r}
img_gut_ko <- fread("data_raw/hmp_img_ko_profile.txt", header = T, sep = "\t")
head(img_gut_ko)
img_gut_ko <- img_gut_ko %>% 
  as.data.frame() 

gut_ko <- gsub("KO:", "", img_gut_ko$`Function Id`)
dbmm_ko <- rownames(ghostko.df)

length(intersect(gut_ko, dbmm_ko))/length(gut_ko) *100

length(which(rowSums(img_gut_ko[,-c(1:2, 103)]) >1))
rownames(img_gut_ko) <- img_gut_ko$`Function Id`
core.KO <- img_gut_ko[,-c(1:2, 103)]
core.KO <- as.data.frame(lapply(core.KO, as.numeric))
rownames(core.KO) <- gsub("KO:", "", img_gut_ko$`Function Id`)
kos.p <- tibble(kos = rownames(core.KO),
                prev = microbiome::prevalence(core.KO, 1, 
                                              count= T, 
                                              include.lowest = F)) %>% 
  filter(prev > 75)

length(intersect(kos.p$kos, dbmm_ko))/length(kos.p$kos) *100
#71.7

kos.p <- tibble(kos = rownames(core.KO),
                prev = microbiome::prevalence(core.KO, 1, 
                                              count= T, 
                                              include.lowest = F)) %>% 
  filter(prev > 50)
length(intersect(kos.p$kos, dbmm_ko))/length(kos.p$kos) *100
# 62.83

kos.p <- tibble(kos = rownames(core.KO),
                prev = microbiome::prevalence(core.KO, 1, 
                                              count= T, 
                                              include.lowest = F)) %>% 
  filter(prev > 90)
length(intersect(kos.p$kos, dbmm_ko))/length(kos.p$kos) *100
#80.12
```


### Reviewer 2 n 3  
Genome potential  

```{r}
library(data.table)
# We need rownames and KO ID
ghostko <- fread("data_raw/04_metatrans/kegg_brite/GhostKola_DbMM_ko_list.txt", header = F, sep = "\t")
#head(ghostko)
colnames(ghostko) <- c("LocusTag", "KO")
ghostko.df <- as.data.frame(ghostko) 

ghostko.df$LocusTag <- stringr::str_extract(ghostko.df$LocusTag, "[^_]*_[^_]*")
ghostko.df <- ghostko.df %>% 
  filter(!is.na(KO)) %>% 
  group_by(LocusTag) %>% 
  distinct(KO) %>% 
  mutate(presence.absence = ifelse(is.na(KO), 0, 1)) %>% 
  pivot_wider(id_cols = KO , names_from = LocusTag, values_from = presence.absence) %>% 
  mutate(across(everything(), .fns = ~replace_na(.,0))) %>% 
  column_to_rownames("KO")
```

GMM ko ids
```{r}

gmmko <- fread("data_raw/05_gmm/ModulesCustom/gmm_core_modules_koids.txt", header = T, sep = "\t")

ghostko.df$KO <- rownames(ghostko.df)
ghostko.df.sub <- filter(ghostko.df, KO %in% gmmko$ko)
dim(ghostko.df)
gmmko <- gmmko %>% 
  left_join(ghostko.df.sub, by = c("ko" = "KO"))
gmmko <- reshape2::melt(gmmko)
head(gmmko)

gmmko$module <- stringr::str_wrap(gmmko$module, width = 15) 
  
```

### Carbs  

```{r}
gmmko %>% 
  filter(type == "Carbohydrates") %>% 
ggplot(aes(ko,variable)) +
  geom_tile(aes(fill=as.factor(value))) +
  facet_grid(~module, scales="free") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90),
        axis.text.y = element_text(face = "italic")) +
  scale_fill_manual("Presence/Absence", values = c("1" = "steelblue", "0" = "grey70"),
                    na.value = 'grey70') +
  labs(x="", y ="")
ggsave("data/04_metatrans/figs/reveiwer_carbs_gmm_genomic_potential.pdf", height =4, width = 10)
```

### Simpe sugars  

```{r}
gmmko %>% 
  filter(type %in% c("Sugar Acid", "SimpleCarbohydrates")) %>% 
ggplot(aes(ko,variable)) +
  geom_tile(aes(fill=as.factor(value))) +
  facet_wrap(~module, scales="free_x") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90),
        axis.text.y = element_text(face = "italic")) +
  scale_fill_manual("Presence/Absence", values = c("1" = "steelblue", "0" = "grey70"),
                    na.value = 'grey70')+
  labs(x="", y ="")
ggsave("data/04_metatrans/figs/reveiwer_simple_gmm_genomic_potential.pdf", height =8, width = 10)
```

### Fermentaation  

```{r}
gmmko %>% 
  filter(type %in% c("Fermentation")) %>% 
ggplot(aes(ko,variable)) +
  geom_tile(aes(fill=as.factor(value))) +
  facet_wrap(~module, scales="free_x") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90),
        axis.text.y = element_text(face = "italic")) +
  scale_fill_manual("Presence/Absence", values = c("1" = "steelblue", "0" = "grey70"),
                    na.value = 'grey70')+
  labs(x="", y ="")
ggsave("data/04_metatrans/figs/reveiwer_fermentation_gmm_genomic_potential.pdf", height =8, width = 10)
```

Group by module.  

```{r}
gmmko.grp <- gmmko %>% 
  group_by(module, ko, variable) %>% 
  mutate()
```



```{r eval=F}
sessionInfo()
```

